--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/etc/init.d/chrony	2022-03-15 08:45:14.000000000 -0400
+++ /media/mks/armbi_root/etc/init.d/chrony	2023-05-08 16:05:00.000000000 -0400
@@ -22,8 +22,7 @@
 DAEMON=/usr/sbin/chronyd
 NAME="chronyd"
 DESC="time daemon"
-PIDFILE=/run/chronyd.pid
-CHRONY_HELPER=/usr/lib/chrony/chrony-helper
+PIDFILE=/run/chrony/chronyd.pid
 
 [ -x "$DAEMON" ] || exit 0
 
@@ -42,16 +41,13 @@
         else
             log_daemon_msg "Starting $DESC" "$NAME"
             start-stop-daemon --start --quiet --pidfile $PIDFILE --exec $DAEMON -- $DAEMON_OPTS
-            if [ -x $CHRONY_HELPER ]; then
-                $CHRONY_HELPER update-daemon
-            fi
             log_end_msg $?
         fi
         ;;
 
     stop)
         log_daemon_msg "Stopping $DESC" "$NAME"
-        start-stop-daemon --stop --quiet --oknodo --pidfile $PIDFILE --remove-pidfile --exec $DAEMON
+        start-stop-daemon --stop --quiet --oknodo --pidfile $PIDFILE --exec $DAEMON
         log_end_msg $?
         ;;
 
