--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/etc/init.d/networking	2018-08-25 10:09:19.000000000 -0400
+++ /media/mks/armbi_root/etc/init.d/networking	2023-01-23 16:14:41.000000000 -0500
@@ -9,13 +9,13 @@
 # Description:       Prepare /run/network directory, ifstate file and raise network interfaces, or take them down.
 ### END INIT INFO
 
-PATH="/sbin:/bin"
+PATH="/sbin:/bin:/usr/sbin:/usr/bin"
 RUN_DIR="/run/network"
 IFSTATE="$RUN_DIR/ifstate"
 STATEDIR="$RUN_DIR/state"
 
-[ -x /sbin/ifup ] || exit 0
-[ -x /sbin/ifdown ] || exit 0
+[ -x "$(command -v ifup)" ] || exit 0
+[ -x "$(command -v ifdown)" ] || exit 0
 
 . /lib/lsb/init-functions
 
@@ -103,9 +103,9 @@
     then
 	    ifaces=$(for iface in $(ifquery --list --allow=hotplug)
 			    do
-				    link=${iface##:*}
-				    link=${link##.*}
-				    if [ -e "/sys/class/net/$link" ]
+				    link=${iface%%:*}
+				    link=${link%%.*}
+				    if [ -e "/sys/class/net/$link" ] && ! ifquery --state "$iface" >/dev/null
 				    then
 					echo "$iface"
 				    fi
@@ -130,9 +130,9 @@
 	set -f
 	exclusions=$(process_exclusions)
 	log_action_begin_msg "Configuring network interfaces"
-	if [ -x /bin/udevadm ]; then
+	if [ -x "$(command -v udevadm)" ]; then
 		if [ -n "$(ifquery --list --exclude=lo)" ] || [ -n "$(ifquery --list --allow=hotplug)" ]; then
-			/bin/udevadm settle || true
+			udevadm settle || true
 		fi
 	fi
 	if ifup -a $exclusions $verbose && ifup_hotplug $exclusions $verbose
