--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/etc/init.d/ssh	2020-01-31 15:55:34.000000000 -0500
+++ /media/mks/armbi_root/etc/init.d/ssh	2023-09-23 18:11:33.000000000 -0400
@@ -1,7 +1,7 @@
 #! /bin/sh
 
 ### BEGIN INIT INFO
-# Provides:		sshd
+# Provides:		ssh sshd
 # Required-Start:	$remote_fs $syslog
 # Required-Stop:	$remote_fs $syslog
 # Default-Start:	2 3 4 5
@@ -68,6 +68,7 @@
 
 check_config() {
     if [ ! -e /etc/ssh/sshd_not_to_be_run ]; then
+	# shellcheck disable=SC2086
 	/usr/sbin/sshd $SSHD_OPTS -t || exit 1
     fi
 }
@@ -80,6 +81,7 @@
 	check_for_no_start
 	check_dev_null
 	log_daemon_msg "Starting OpenBSD Secure Shell server" "sshd" || true
+	# shellcheck disable=SC2086
 	if start-stop-daemon --start --quiet --oknodo --chuid 0:0 --pidfile /run/sshd.pid --exec /usr/sbin/sshd -- $SSHD_OPTS; then
 	    log_end_msg 0 || true
 	else
@@ -113,6 +115,7 @@
 	start-stop-daemon --stop --quiet --oknodo --retry 30 --pidfile /run/sshd.pid --exec /usr/sbin/sshd
 	check_for_no_start log_end_msg
 	check_dev_null log_end_msg
+	# shellcheck disable=SC2086
 	if start-stop-daemon --start --quiet --oknodo --chuid 0:0 --pidfile /run/sshd.pid --exec /usr/sbin/sshd -- $SSHD_OPTS; then
 	    log_end_msg 0 || true
 	else
@@ -131,6 +134,7 @@
 		# old daemon stopped
 		check_for_no_start log_end_msg
 		check_dev_null log_end_msg
+		# shellcheck disable=SC2086
 		if start-stop-daemon --start --quiet --oknodo --chuid 0:0 --pidfile /run/sshd.pid --exec /usr/sbin/sshd -- $SSHD_OPTS; then
 		    log_end_msg 0 || true
 		else
