--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/home/mks/moonraker/scripts/fetch-apikey.sh	2022-07-25 05:53:30.922869338 -0400
+++ /media/mks/armbi_root/home/mks/moonraker/scripts/fetch-apikey.sh	2023-12-13 12:36:32.477233867 -0500
@@ -1,6 +1,6 @@
 #!/bin/bash
 # Helper Script for fetching the API Key from a moonraker database
-DATABASE_PATH="${HOME}/.moonraker_database"
+DATABASE_PATH="${HOME}/printer_data/database"
 MOONRAKER_ENV="${HOME}/moonraker-env"
 DB_ARGS="--read=READ --db=authorized_users get _API_KEY_USER_"
 API_REGEX='(?<="api_key": ")([^"]+)'
