--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/home/mks/moonraker-env/bin/normalizer	2022-07-25 05:57:33.908115840 -0400
+++ /media/mks/armbi_root/home/mks/moonraker-env/bin/normalizer	2023-12-13 12:43:40.121996367 -0500
@@ -1,8 +1,8 @@
-#!/home/mks/moonraker-env/bin/python3
+#!/home/mks/moonraker-env/bin/python
 # -*- coding: utf-8 -*-
 import re
 import sys
-from charset_normalizer.cli.normalizer import cli_detect
+from charset_normalizer.cli import cli_detect
 if __name__ == '__main__':
     sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
     sys.exit(cli_detect())
