--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/home/mks/moonraker/.git/hooks/pre-commit.sample	2022-07-25 05:53:05.534774198 -0400
+++ /media/mks/armbi_root/home/mks/moonraker/.git/hooks/pre-commit.sample	2023-12-13 12:36:29.181178497 -0500
@@ -16,7 +16,7 @@
 fi
 
 # If you want to allow non-ASCII filenames set this variable to true.
-allownonascii=$(git config --bool hooks.allownonascii)
+allownonascii=$(git config --type=bool hooks.allownonascii)
 
 # Redirect output to stderr.
 exec 1>&2
