--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/lib/mc/ext.d/sound.sh	2019-01-18 08:10:34.000000000 -0500
+++ /media/mks/armbi_root/usr/lib/mc/ext.d/sound.sh	2023-02-14 07:56:55.000000000 -0500
@@ -12,6 +12,10 @@
     filetype=$1
 
     case "${filetype}" in
+    common)
+        mediainfo "${MC_EXT_FILENAME}"
+        ;;
+
     mp3)
         mpg123 -vtn1 "${MC_EXT_FILENAME}" 2>&1 | \
             sed -n '/^Title/,/^Comment/p;/^MPEG/,/^Audio/p'
@@ -19,6 +23,9 @@
     ogg)
         ogginfo "${MC_EXT_FILENAME}"
         ;;
+    opus)
+        opusinfo "${MC_EXT_FILENAME}"
+        ;;
     wma)
         mplayer -quiet -slave -frames 0 -vo null -ao null -identify "${MC_EXT_FILENAME}" 2>/dev/null | \
             tail +13 || file "${MC_EXT_FILENAME}"
@@ -35,7 +42,7 @@
     case "${filetype}" in
     common)
         if [ -n "$DISPLAY" ]; then
-            (xmms  "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
+            (audacious  "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
         else
             play "${MC_EXT_FILENAME}"
         fi
@@ -49,18 +56,25 @@
         ;;
     mp3)
         if [ -n "$DISPLAY" ]; then
-            (xmms "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
+            (audacious "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
         else
             mpg123 "${MC_EXT_FILENAME}"
         fi
         ;;
     ogg)
         if [ -n "$DISPLAY" ]; then
-            (xmms "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
+            (audacious "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
         else
             ogg123 "${MC_EXT_FILENAME}"
         fi
         ;;
+    opus)
+        if [ -n "$DISPLAY" ]; then
+            (audacious "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
+        else
+            play "${MC_EXT_FILENAME}"
+        fi
+        ;;
     midi)
         timidity "${MC_EXT_FILENAME}"
         ;;
@@ -69,7 +83,7 @@
         ;;
     playlist)
         if [ -n "$DISPLAY" ]; then
-            (xmms -p "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
+            (audacious -p "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
         else
             mplayer -vo null -playlist "${MC_EXT_FILENAME}"
         fi
