--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/lib/armbian/armbian-truncate-logs	2022-07-21 02:19:01.000000000 -0400
+++ /media/mks/armbi_root/usr/lib/armbian/armbian-truncate-logs	2023-12-12 11:04:09.000000000 -0500
@@ -16,8 +16,14 @@
 
 [ "$ENABLED" != true ] && exit 0
 
+if [ -z "`grep ^Storage=volatile /etc/systemd/journald.conf`" -a -z "$(ls -A /run/log/journal)" ] ;then
+    journalctl --relinquish-var # make sure to do volatile logging until the next flush
+fi
+
 logusage=$(df /var/log/ --output=pcent | tail -1 |cut -d "%" -f 1)
 if [ $logusage -ge $treshold ]; then
+    # remove journal backup files created by armbian-ramdisk in case of duplicate journals directories
+    rm -rf /var/log.hdd/journal-*
     # write to SD
     /usr/lib/armbian/armbian-ramlog write >/dev/null 2>&1
     # rotate logs on "disk"
@@ -29,5 +35,7 @@
     # remove
     /usr/bin/find /var/log -name '*.[0-9]' -or -name '*.gz' | xargs -r rm -f
     # vacuum systemd-journald
-    [ -d /var/log/journal ] && journalctl --vacuum-size=${JOURNAL_SIZE}
+    [ -d /var/log/journal ] && journalctl --quiet --vacuum-size=${JOURNAL_SIZE}
+    # remove old archived journal files modified more than 1 day ago
+    [ -d /var/log.hdd/journal ] && find /var/log.hdd/journal -ctime 1 \( -name '*@*' -o -name '*~' \)  -delete
 fi
