--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/bin/encguess	2020-07-21 15:27:00.000000000 -0400
+++ /media/mks/armbi_root/usr/bin/encguess	2023-11-25 15:59:54.000000000 -0500
@@ -1,6 +1,6 @@
 #!/usr/bin/perl
     eval 'exec /usr/bin/perl -S $0 ${1+"$@"}'
-	if $running_under_some_shell;
+	if 0; # ^ Run only under a shell
 #!./perl
 use 5.008001;
 BEGIN { pop @INC if $INC[-1] eq '.' }
@@ -64,7 +64,7 @@
 
 =head1 VERSION
 
-$Id: encguess,v 0.2 2016/08/04 03:15:58 dankogai Exp $
+$Id: encguess,v 0.3 2020/12/02 01:28:17 dankogai Exp $
 
 =head1 SYNOPSIS
 
@@ -81,7 +81,7 @@
 =item -s
 
 specify a list of "suspect encoding types" to test, 
-seperated by either C<:> or C<,>
+separated by either C<:> or C<,>
 
 =item -S
 
