--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/lib/ifupdown/wait-online.sh	2019-01-28 15:37:33.000000000 -0500
+++ /media/mks/armbi_root/usr/lib/ifupdown/wait-online.sh	2023-01-24 08:07:32.000000000 -0500
@@ -11,7 +11,7 @@
 case "$WAIT_ONLINE_METHOD" in
 route)
 	[ -n "$WAIT_ONLINE_ADDRESS" ] || WAIT_ONLINE_ADDRESS=default
-	(/usr/bin/timeout "$WAIT_ONLINE_TIMEOUT" /sbin/ip mon r & /sbin/ip -4 r s; /sbin/ip -6 r s) | /bin/grep -q "^$WAIT_ONLINE_ADDRESS\>"
+	(timeout "$WAIT_ONLINE_TIMEOUT" ip mon r & ip -4 r s; ip -6 r s) | grep -q "^$WAIT_ONLINE_ADDRESS\>"
 	;;
 
 ping)
@@ -19,23 +19,27 @@
 		echo "No WAIT_ONLINE_ADDRESS specified" >&2
 		exit 1
 	fi
-	/bin/ping -q -c 1 -w "$WAIT_ONLINE_TIMEOUT" "$WAIT_ONLINE_ADDRESS" >/dev/null
+	timeout "$WAIT_ONLINE_TIMEOUT" sh -c 'while ! ping -q -c 1 -W 1 "'$WAIT_ONLINE_ADDRESS'" >/dev/null; do sleep 1; done'
 	;;
 
 ping6)
-	/bin/ping6 -q -c 1 -w "$WAIT_ONLINE_TIMEOUT" "$WAIT_ONLINE_ADDRESS" >/dev/null
+	if [ -z "$WAIT_ONLINE_ADDRESS" ]; then
+		echo "No WAIT_ONLINE_ADDRESS specified" >&2
+		exit 1
+	fi
+	timeout "$WAIT_ONLINE_TIMEOUT" sh -c 'while ! ping6 -q -c 1 -W 1 "'$WAIT_ONLINE_ADDRESS'" >/dev/null; do sleep 1; done'
 	;;
 
 ifup|iface|interface)
 	up=false
 	if [ -z "$WAIT_ONLINE_IFACE" ]; then
-		auto_list="$(/sbin/ifquery -X lo --list)"
-		hotplug_list="$(/sbin/ifquery -X lo --allow=hotplug --list)"
+		auto_list="$(ifquery -X lo --list)"
+		hotplug_list="$(ifquery -X lo --allow=hotplug --list)"
 		if [ -n "$auto_list" ]; then
 			for i in $(seq 1 $WAIT_ONLINE_TIMEOUT); do
 				up=true
 				for iface in $auto_list; do
-					if ! /sbin/ifquery --state $iface >/dev/null; then
+					if ! ifquery --state $iface >/dev/null; then
 						up=false
 						break
 					fi
@@ -47,7 +51,7 @@
 			done
 		elif [ -n "$hotplug_list" ]; then
 			for i in $(seq 1 $WAIT_ONLINE_TIMEOUT); do
-				if [ -n "$(/sbin/ifquery --state $hotplug_list)" ]; then
+				if [ -n "$(ifquery --state $hotplug_list)" ]; then
 					up=true
 					break
 				fi
@@ -58,7 +62,7 @@
 		fi
 	else
 		for i in $(seq 1 $WAIT_ONLINE_TIMEOUT); do
-			if [ -n "$(/sbin/ifquery --state $WAIT_ONLINE_IFACE)" ]; then
+			if [ -n "$(ifquery --state $WAIT_ONLINE_IFACE)" ]; then
 				up=true
 				break
 			fi
