--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/lib/openssh/agent-launch	2020-01-31 15:55:34.000000000 -0500
+++ /media/mks/armbi_root/usr/lib/openssh/agent-launch	2023-12-19 07:55:09.000000000 -0500
@@ -3,6 +3,7 @@
 set -e
 
 if [ ! -d "$XDG_RUNTIME_DIR" ]; then
+    # shellcheck disable=SC2016
     echo 'This needs $XDG_RUNTIME_DIR to be set' >&2
     exit 1
 fi
@@ -10,8 +11,8 @@
 if [ "$1" = start ]; then
     if [ -z "$SSH_AUTH_SOCK" ] && grep -s -q '^use-ssh-agent$' /etc/X11/Xsession.options; then
         S="$XDG_RUNTIME_DIR/openssh_agent"
-        dbus-update-activation-environment --verbose --systemd SSH_AUTH_SOCK=$S SSH_AGENT_LAUNCHER=openssh
-        exec ssh-agent -D -a $S
+        dbus-update-activation-environment --verbose --systemd SSH_AUTH_SOCK="$S" SSH_AGENT_LAUNCHER=openssh
+        exec ssh-agent -D -a "$S"
     fi
 elif [ "$1" = stop ]; then
     if [ "$SSH_AGENT_LAUNCHER" = openssh ]; then
