--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/share/doc/dialog/examples/form1	2011-10-04 19:36:53.000000000 -0400
+++ /media/mks/armbi_root/usr/share/doc/dialog/examples/form1	2020-11-25 19:34:39.000000000 -0500
@@ -1,5 +1,5 @@
 #! /bin/sh
-# $Id: form1,v 1.15 2011/10/04 23:36:53 tom Exp $
+# $Id: form1,v 1.18 2020/11/26 00:34:39 tom Exp $
 
 . ./setup-vars
 
@@ -19,7 +19,7 @@
 while test $returncode != 1 && test $returncode != 250
 do
 exec 3>&1
-value=`$DIALOG --ok-label "Submit" \
+returntext=`$DIALOG --ok-label "Submit" \
 	  --backtitle "$backtitle" "$@" \
 	  --form "Here is a possible piece of a configuration program." \
 20 50 0 \
@@ -31,7 +31,7 @@
 returncode=$?
 exec 3>&-
 
-show=`echo "$value" |sed -e 's/^/	/'`
+show=`echo "$returntext" |sed -e 's/^/	/'`
 
 	case $returncode in
 	$DIALOG_CANCEL)
@@ -63,16 +63,8 @@
 		echo "Button 3 (Extra) pressed."
 		exit
 		;;
-	$DIALOG_ERROR)
-		echo "ERROR!$value"
-		exit
-		;;
-	$DIALOG_ESC)
-		echo "ESC pressed."
-		exit
-		;;
 	*)
-		echo "Return code was $returncode"
+		. ./report-button
 		exit
 		;;
 	esac
