--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/bin/corelist	2020-07-21 15:27:00.000000000 -0400
+++ /media/mks/armbi_root/usr/bin/corelist	2023-11-25 15:59:54.000000000 -0500
@@ -1,6 +1,6 @@
 #!/usr/bin/perl
     eval 'exec /usr/bin/perl -S $0 ${1+"$@"}'
-	if $running_under_some_shell;
+	if 0; # ^ Run only under a shell
 #!/usr/bin/perl
 
 =head1 NAME
@@ -227,7 +227,15 @@
     my ($old_ver, $new_ver) = @ARGV;
 
     my $old = numify_version($old_ver);
+    if ( !Module::CoreList->find_version($old) ) {
+        print "\nModule::CoreList has no info on perl $old_ver\n\n";
+        exit 1;
+    }
     my $new = numify_version($new_ver);
+    if ( !Module::CoreList->find_version($new) ) {
+        print "\nModule::CoreList has no info on perl $new_ver\n\n";
+        exit 1;
+    }
 
     my %diff = Module::CoreList::changes_between($old, $new);
 
