--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/bin/znew	2022-04-15 14:16:55.000000000 -0400
+++ /media/mks/armbi_root/usr/bin/znew	2022-04-09 22:22:26.000000000 -0400
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-# Copyright (C) 1998, 2002, 2004, 2007, 2010-2018 Free Software Foundation,
+# Copyright (C) 1998, 2002, 2004, 2007, 2010-2022 Free Software Foundation,
 # Inc.
 # Copyright (C) 1993 Jean-loup Gailly
 
@@ -21,8 +21,8 @@
 LC_ALL=C
 export LC_ALL
 
-version="znew (gzip) 1.9
-Copyright (C) 2010-2017 Free Software Foundation, Inc.
+version="znew (gzip) 1.12
+Copyright (C) 2010-2018 Free Software Foundation, Inc.
 This is free software.  You may redistribute copies of it under the terms of
 the GNU General Public License <https://www.gnu.org/licenses/gpl.html>.
 There is NO WARRANTY, to the extent permitted by law.
