--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/share/doc/dialog/examples/calendar-stdout	2010-01-13 05:20:03.000000000 -0500
+++ /media/mks/armbi_root/usr/share/doc/dialog/examples/calendar-stdout	2020-11-25 19:09:12.000000000 -0500
@@ -1,9 +1,9 @@
 #!/bin/sh
-# $Id: calendar-stdout,v 1.6 2010/01/13 10:20:03 tom Exp $
+# $Id: calendar-stdout,v 1.8 2020/11/26 00:09:12 tom Exp $
 
 . ./setup-vars
 
-RESULT=`$DIALOG --stdout --title "CALENDAR" "$@" --calendar "Please choose a date..." 0 0 7 7 1981`
-retval=$?
+returntext=`$DIALOG --stdout --title "CALENDAR" "$@" --calendar "Please choose a date..." 0 0 7 7 1981`
+returncode=$?
 
 . ./report-string
