--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/bin/lcf	2018-12-14 03:51:14.000000000 -0500
+++ /media/mks/armbi_root/usr/bin/lcf	2023-01-27 08:29:51.000000000 -0500
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #                               -*- Mode: Sh -*-
 # lcf ---
 # Author           : Manoj Srivastava ( srivasta@glaurung.green-gryphon.com )
@@ -206,7 +206,7 @@
     if [ -d "$old_mdsum_dir" ]; then
 	echo "The historical md5sum dir $old_mdsum_dir exists"
     elif [ -f "$old_mdsum_file" ]; then
-	echo "The histotical md5sum file $old_mdsum_file exists"
+	echo "The historical md5sum file $old_mdsum_file exists"
     else
 	echo "Historical md5sums are not available"
     fi
@@ -238,7 +238,7 @@
 #	   (like just one), the maintainer is also making a guess at
 #	   this point by supplying a historical md5sum default file.
     if [ "X$VERBOSE" != "X" ]; then
-	echo >&2 "Histotical md5sums did not match."
+	echo >&2 "Historical md5sums did not match."
     fi
     if [ -d "$old_mdsum_dir"  ]; then
 	if [ -e "${old_mdsum_dir}/default" ]; then
