--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/share/doc/dialog/examples/fselect1-stdout	2010-01-13 05:20:03.000000000 -0500
+++ /media/mks/armbi_root/usr/share/doc/dialog/examples/fselect1-stdout	2020-11-25 19:09:12.000000000 -0500
@@ -1,18 +1,18 @@
 #!/bin/sh
-# $Id: fselect1-stdout,v 1.6 2010/01/13 10:20:03 tom Exp $
+# $Id: fselect1-stdout,v 1.9 2020/11/26 00:09:12 tom Exp $
 
 . ./setup-vars
 
 FILE=$HOME
 for n in .cshrc .profile .bashrc
 do
-	if test -f $HOME/$n ; then
+	if test -f "$HOME/$n" ; then
 		FILE=$HOME/$n
 		break
 	fi
 done
 
-RESULT=`$DIALOG --stdout --title "Please choose a file" "$@" --fselect $FILE 14 48`
-retval=$?
+returntext=`$DIALOG --stdout --title "Please choose a file" "$@" --fselect "$FILE" 14 48`
+returncode=$?
 
 . ./report-string
