--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/share/initramfs-tools/hooks/udev	2021-07-08 09:00:48.000000000 -0400
+++ /media/mks/armbi_root/usr/share/initramfs-tools/hooks/udev	2024-06-16 05:44:13.000000000 -0400
@@ -22,16 +22,16 @@
 
 # copy .link files containing interface naming definitions
 mkdir -p "$DESTDIR/lib/systemd/network/"
-find /lib/systemd/network -name '*.link' -execdir cp -pt "$DESTDIR/lib/systemd/network/" '{}' +
+find -L /lib/systemd/network -name '*.link' -execdir cp -pt "$DESTDIR/lib/systemd/network/" '{}' +
 if [ -d /etc/systemd/network ]; then
-  find /etc/systemd/network -name '*.link' -execdir cp -pt "$DESTDIR/lib/systemd/network/" '{}' +
+  find -L /etc/systemd/network -name '*.link' -execdir cp -pt "$DESTDIR/lib/systemd/network/" '{}' +
 fi
 
 mkdir -p "$DESTDIR/lib/udev/rules.d/"
 for rules in 50-firmware.rules 50-udev-default.rules \
         60-block.rules 60-persistent-storage.rules \
-        61-persistent-storage-android.rules 71-seat.rules 73-special-net-names.rules \
-        73-usb-net-by-mac.rules 75-net-description.rules \
+        61-persistent-storage-android.rules 71-seat.rules \
+        73-special-net-names.rules 75-net-description.rules \
         80-net-setup-link.rules 80-drivers.rules; do
   if   [ -e /etc/udev/rules.d/$rules ]; then
     cp -p /etc/udev/rules.d/$rules "$DESTDIR/lib/udev/rules.d/"
