--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/lib/udev/hwclock-set	2019-01-10 03:30:43.000000000 -0500
+++ /media/mks/armbi_root/usr/lib/udev/hwclock-set	2024-03-28 05:52:12.000000000 -0400
@@ -8,30 +8,5 @@
     exit 0
 fi
 
-if [ -e /run/udev/hwclock-set ]; then
-    exit 0
-fi
-
-if [ -f /etc/default/rcS ] ; then
-    . /etc/default/rcS
-fi
-
-# These defaults are user-overridable in /etc/default/hwclock
-BADYEAR=no
-HWCLOCKACCESS=yes
-HWCLOCKPARS=
-HCTOSYS_DEVICE=rtc0
-if [ -f /etc/default/hwclock ] ; then
-    . /etc/default/hwclock
-fi
-
-if [ yes = "$BADYEAR" ] ; then
-    /sbin/hwclock --rtc=$dev --systz --badyear
-    /sbin/hwclock --rtc=$dev --hctosys --badyear
-else
-    /sbin/hwclock --rtc=$dev --systz
-    /sbin/hwclock --rtc=$dev --hctosys
-fi
-
-# Note 'touch' may not be available in initramfs
-> /run/udev/hwclock-set
+/sbin/hwclock --rtc=$dev --systz
+/sbin/hwclock --rtc=$dev --hctosys
