--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/lib/mc/extfs.d/uzip	2019-01-18 08:10:34.000000000 -0500
+++ /media/mks/armbi_root/usr/lib/mc/extfs.d/uzip	2023-02-14 07:56:55.000000000 -0500
@@ -352,7 +352,7 @@
 	if ($platform eq 'unx' && $filename =~ /\/$/ && $perms =~ /^\?(.*)$/) {
 		$perms = 'd'.$1;
 	}
-	printf "%-10s    1 %-8d %-8d %8s %s/%s/%s %s:%s:%s %s", $perms, $<,
+	printf "%-10s    1 %-8d %-8d %8s %s/%s/%s %s:%s:%s ./%s", $perms, $<,
 		$(, $realsize, $mon, $day, $year, $hours, $mins, $secs, $filename;
 	if ($platform eq 'unx' && $perms =~ /^l/) {
 		my $linkdest = &get_link_destination($filename);
