--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/share/bug/apt/script	2021-04-19 12:41:13.000000000 -0400
+++ /media/mks/armbi_root/usr/share/bug/apt/script	2023-05-25 10:11:37.000000000 -0400
@@ -1,9 +1,5 @@
-#!/bin/bash -e
-
-# reportbug #169495
-if [ -z "$YESNO" ]; then
-  YESNO=$"yYnN"
-fi
+#!/bin/bash
+set -e
 
 cat <<EOF
 I can automatically include various information about your apt configuration in
@@ -14,20 +10,20 @@
 yesno "May I include your apt configuration (/etc/apt/apt.conf et al)? [Y/n] " yep
 
 if [ "$REPLY" = "yep" ]; then
-  echo -e "\n-- apt-config dump --\n" >&3
+  printf "\n-- apt-config dump --\n\n" >&3
   apt-config dump >&3 2>&1
 fi
 
 for config in /etc/apt/preferences /etc/apt/preferences.d/* /etc/apt/sources.list /etc/apt/sources.list.d/* ; do
-  if [ -f $config ]; then
+  if [ -f "$config" ]; then
     yesno "May I include your $config configuration file? [Y/n] " yep
     if [ "$REPLY" = "yep" ]; then
-      echo -e "\n-- $config --\n" >&3
-      cat $config >&3
+      printf "\n-- %s --\n\n" "$config" >&3
+      cat "$config" >&3
     else
-      echo -e "\n-- ($config present, but not submitted) --\n" >&3
+      printf "\n-- (%s present, but not submitted) --\n\n" "$config" >&3
     fi
   else
-    echo -e "\n-- (no $config present) --\n" >&3
+    printf "\n-- (no %s present) --\n\n" "$config" >&3
   fi
 done
