--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/bin/ifnames	2017-08-20 14:17:16.000000000 -0400
+++ /media/mks/armbi_root/usr/bin/ifnames	2023-01-14 12:42:30.000000000 -0500
@@ -1,13 +1,13 @@
-#! /usr/bin/perl -w
+#! /usr/bin/perl
 # -*- perl -*-
-# Generated from ifnames.in; do not edit by hand.
+# Generated from bin/ifnames.in; do not edit by hand.
 
 eval 'case $# in 0) exec /usr/bin/perl -S "$0";; *) exec /usr/bin/perl -S "$0" "$@";; esac'
     if 0;
 
 # ifnames - print the identifiers used in C preprocessor conditionals
 
-# Copyright (C) 1994-1995, 1999-2003, 2005-2012 Free Software
+# Copyright (C) 1994-1995, 1999-2003, 2005-2017, 2020-2021 Free Software
 # Foundation, Inc.
 
 # This program is free software: you can redistribute it and/or modify
@@ -21,7 +21,7 @@
 # GNU General Public License for more details.
 
 # You should have received a copy of the GNU General Public License
-# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 # Reads from stdin if no files are given.
 # Writes to stdout.
@@ -29,6 +29,10 @@
 # Written by David MacKenzie <djm@gnu.ai.mit.edu>
 # and Paul Eggert <eggert@twinsun.com>.
 
+use 5.006;
+use strict;
+use warnings FATAL => 'all';
+
 BEGIN
 {
   my $pkgdatadir = $ENV{'autom4te_perllibdir'} || '/usr/share/autoconf';
@@ -38,7 +42,7 @@
   # that can handle redirection and quote arguments correctly,
   # e.g.: COMMAND.COM.  For DJGPP always use the shell that configure
   # has detected.
-  $ENV{'SHELL'} = '/bin/bash' if ($^O eq 'dos');
+  $ENV{'SHELL'} = '/bin/sh' if ($^O eq 'dos');
 }
 
 use Autom4te::General;
@@ -51,25 +55,25 @@
 
 Scan all of the C source FILES (or the standard input, if none are
 given) and write to the standard output a sorted list of all the
-identifiers that appear in those files in `#if', `#elif', `#ifdef', or
-`#ifndef' directives.  Print each identifier on a line, followed by a
+identifiers that appear in those files in '#if', '#elif', '#ifdef', or
+'#ifndef' directives.  Print each identifier on a line, followed by a
 space-separated list of the files in which that identifier occurs.
 
   -h, --help      print this help, then exit
   -V, --version   print version number, then exit
 
 Report bugs to <bug-autoconf\@gnu.org>.
-GNU Autoconf home page: <http://www.gnu.org/software/autoconf/>.
-General help using GNU software: <http://www.gnu.org/gethelp/>.
+GNU Autoconf home page: <https://www.gnu.org/software/autoconf/>.
+General help using GNU software: <https://www.gnu.org/gethelp/>.
 ";
 
 
 # $VERSION
 # --------
-$version = "ifnames (GNU Autoconf) 2.69
-Copyright (C) 2012 Free Software Foundation, Inc.
+$version = "ifnames (GNU Autoconf) 2.71
+Copyright (C) 2021 Free Software Foundation, Inc.
 License GPLv3+/Autoconf: GNU GPL version 3 or later
-<http://gnu.org/licenses/gpl.html>, <http://gnu.org/licenses/exceptions.html>
+<https://gnu.org/licenses/gpl.html>, <https://gnu.org/licenses/exceptions.html>
 This is free software: you are free to change and redistribute it.
 There is NO WARRANTY, to the extent permitted by law.
 
@@ -96,7 +100,7 @@
 sub scan_file ($)
 {
   my ($file_name) = @_;
-  my $file = new Autom4te::XFile ("< " . open_quote ($file_name));
+  my $file = new Autom4te::XFile ($file_name, "<");
   while ($_ = $file->getline)
     {
       # Continuation lines.
