--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/sbin/ddbr	2022-07-21 02:12:25.000000000 -0400
+++ /media/mks/armbi_root/usr/sbin/ddbr	2023-12-12 11:04:09.000000000 -0500
@@ -207,7 +207,7 @@
 	echo "$_r UNABLE TO FIND ANY DRIVES ON THIS SYSTEM!!! $_x"
 	exit 1
 fi
-runfrom=$(lsblk | grep /$ | grep -oE '(mmcblk[0-9]|sda[0-9])')
+runfrom=$(lsblk | grep / | grep -oE '(mmcblk[0-9]|sda[0-9])')
 if [ "$runfrom" = "" ]
 then
 	echo "$_r UNABLE TO FIND ROOT OF THE RUNNING SYSTEM!!! $_x"
@@ -235,7 +235,7 @@
 	exit 1
 fi
 intsize=$(fdisk -s /dev/$emmc)
-#image=$(echo $(cat /proc/cpuinfo | egrep '(Hardware|Revision)' | awk '{print $3}') | sed "s/ /-/g")-emmc.img
+#image=$(echo $(grep -E '(Hardware|Revision)' /proc/cpuinfo | awk '{print $3}') | sed "s/ /-/g")-emmc.img
 image=BACKUP-arm-64-emmc.img
 
 ### BACKUP ONLY CHECKS
