--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/lib/mc/ext.d/video.sh	2019-01-18 08:10:34.000000000 -0500
+++ /media/mks/armbi_root/usr/lib/mc/ext.d/video.sh	2023-02-14 07:56:55.000000000 -0500
@@ -13,7 +13,7 @@
 
     case "${filetype}" in
     *)
-        if mplayer >/dev/null 2>&1; then
+        if which mplayer >/dev/null 2>&1; then
             mplayer -identify -vo null -ao null -frames 0 "${MC_EXT_FILENAME}" 2>&1 | \
                 sed -n 's/^ID_//p'
         elif which mpv_identify.sh >/dev/null 2>&1; then
@@ -28,18 +28,22 @@
 do_open_action() {
     filetype=$1
 
+    if which mpv >/dev/null 2>&1; then
+        PLAYER=mpv
+    elif which mplayer >/dev/null 2>&1; then
+        PLAYER=mplayer
+    else
+        echo "Please install either mplayer or mpv to play this file"
+        return
+    fi
+
     case "${filetype}" in
-    ram|rm)
-        (see "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
-        ;;
     *)
         if [ -n "$DISPLAY" ]; then
-            (mplayer "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
+            ($PLAYER "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
         else
-            mplayer -vo null "${MC_EXT_FILENAME}"
+            $PLAYER -vo null "${MC_EXT_FILENAME}"
         fi
-        #(gtv "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
-        #(xanim "${MC_EXT_FILENAME}" >/dev/null 2>&1 &)
         ;;
     esac
 }
