--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/bin/ucf	2018-12-14 03:51:14.000000000 -0500
+++ /media/mks/armbi_root/usr/bin/ucf	2023-01-27 08:29:51.000000000 -0500
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #                               -*- Mode: Sh -*-
 # updateConfFile.sh ---
 # Author           : Manoj Srivastava ( srivasta@glaurung.green-gryphon.com )
@@ -82,8 +82,8 @@
     local new_file="$4"
 
     # Note: get_file_metadata not in quotes to ignore "\n" characters
-    local old_file_label=$(get_file_metadata "$old_file")
-    local new_file_label=$(get_file_metadata "$new_file")
+    local old_file_label="$(get_file_metadata "$old_file")"
+    local new_file_label="$(get_file_metadata "$new_file")"
 
     [ -e "$old_file" ] || old_file=/dev/null
     [ -e "$new_file" ] || new_file=/dev/null
@@ -159,6 +159,7 @@
      -d[n], --debug=[n]      Set the Debug level to N. Please note there must
                              be no spaces before the debug level
      -n,     --no-action     Dry run. No action is actually taken.
+     -P foo, --package foo   Don't follow dpkg-divert diversions by package foo.
      -v,     --verbose       Make the script verbose
              --three-way     Register this file in the cache, and turn on the
                              diff3 option allowing the merging of maintainer
@@ -207,12 +208,12 @@
 	if [ "X$docmd" = "XYES" ]; then
 	    set +e
 	    if [ "X$VERBOSE" != "X" ]; then
-		echo >&2 "egrep -v [[:space:]]${safe_dest_file}$ $statedir/hashfile"
-		egrep -v "[[:space:]]${safe_dest_file}$"  "$statedir/hashfile" >&2 \
+		echo >&2 "grep -Ev [[:space:]]${safe_dest_file}$ $statedir/hashfile"
+		grep -Ev "[[:space:]]${safe_dest_file}$"  "$statedir/hashfile" >&2 \
 		    || true;
 	    fi
-	    #echo "egrep -v [[:space:]]${safe_dest_file}$ $statedir/hashfile"
-	    egrep -v "[[:space:]]${safe_dest_file}$" "$statedir/hashfile" > \
+	    #echo "grep -Ev [[:space:]]${safe_dest_file}$ $statedir/hashfile"
+	    grep -Ev "[[:space:]]${safe_dest_file}$" "$statedir/hashfile" > \
 		"$statedir/hashfile.tmp" || true;
 	    if [ "X$docmd" = "XYES" ]; then
 		mv -f "$statedir/hashfile.tmp"  "$statedir/hashfile"
@@ -249,18 +250,18 @@
 	if [ "X$docmd" = "XYES" ]; then
 	    set +e
 	    if [ "X$VERBOSE" != "X" ]; then
-		echo >&2 "(egrep -v \"[[:space:]]${safe_dest_file}$\" \"$statedir/hashfile\";"
-		egrep -v "[[:space:]]${safe_dest_file}$"  "$statedir/hashfile" >&2 || true;
+		echo >&2 "(grep -Ev \"[[:space:]]${safe_dest_file}$\" \"$statedir/hashfile\";"
+		grep -Ev "[[:space:]]${safe_dest_file}$"  "$statedir/hashfile" >&2 || true;
 		md5sum "$orig_new_file" | sed "s|$orig_new_file|$dest_file|" >&2;
 	    fi
-	    egrep -v "[[:space:]]${safe_dest_file}$" "$statedir/hashfile" > \
+	    grep -Ev "[[:space:]]${safe_dest_file}$" "$statedir/hashfile" > \
 		"$statedir/hashfile.tmp" || true;
 	    md5sum "$orig_new_file" | sed "s|$orig_new_file|$dest_file|" >> \
 		"$statedir/hashfile.tmp";
 	    mv -f "$statedir/hashfile.tmp"  "$statedir/hashfile"
 	    set -e
 	else
-	    echo "(egrep -v \"[[:space:]]${safe_dest_file}$\" \"$statedir/hashfile\""
+	    echo "(grep -Ev \"[[:space:]]${safe_dest_file}$\" \"$statedir/hashfile\""
 	    echo " md5sum \"$orig_new_file\" | sed \"s|$orig_new_file|$dest_file|\"; "
 	    echo ") | sort > \"$statedir/hashfile\""
 	fi
@@ -299,16 +300,22 @@
 		echo >&2 "Not saving ${real_file}, since it was unmodified"
 	    fi
 	else
-	    $action cp -pf "${real_file}" "${real_file}.${OLD_SUFFIX}"
+	    $action cp -pf $selinux "${real_file}" "${real_file}.${OLD_SUFFIX}"
 	fi
     fi
-    $action cp -pf "$new_file" "${real_file}"
+    if [ -e "${real_file}" ]; then
+        # Do not change the permissions and attributes of the destination
+        $action cp -f $selinux "$new_file" "${real_file}"
+    else
+        # No destination file exists
+        $action cp -pf $selinux "$new_file" "${real_file}"
+    fi
     replace_md5sum;
 }
 
 # Escape single quotes in the arguments passed in
 quote_single() {
-    echo "$1" | sed -e "s,','\\\\'',g"
+    printf "%s\n" "$1" | sed -e "s,','\\\\'',g"
 }
 
 
@@ -324,6 +331,7 @@
 docmd='YES'
 action='withecho'
 action=
+selinux=''
 DEBUG=0
 VERBOSE=''
 statedir='/var/lib/ucf';
@@ -341,8 +349,8 @@
 # Note that we use `"$@"' to let each command-line parameter expand to a
 # separate word. The quotes around `$@' are essential!
 # We need TEMP as the `eval set --' would nuke the return value of getopt.
-TEMP=$(getopt -a -o hs:d::D::nv -n "$progname" \
-      --long help,src-dir:,sum-file:,dest-dir:,debug::,DEBUG::,no-action,purge,verbose,three-way,debconf-ok,debconf-template:,state-dir: \
+TEMP=$(getopt -a -o hs:d::D::npP:Zv -n "$progname" \
+      --long help,src-dir:,sum-file:,dest-dir:,debug::,DEBUG::,no-action,package:,purge,verbose,three-way,debconf-ok,debconf-template:,state-dir: \
              -- "$@")
 
 # Note the quotes around `$TEMP': they are essential!
@@ -353,6 +361,8 @@
 	-h|--help) usageversion;                        exit 0 ;;
 	-n|--no-action) action='echo'; docmd='NO';      shift  ;;
 	-v|--verbose) VERBOSE=1;                        shift  ;;
+	-P|--package)
+	    opt_package="$2";			       shift 2 ;;
 	-s|--src-dir)
 	    opt_source_dir="$2";                       shift 2 ;;
 	--sum-file)
@@ -372,6 +382,7 @@
         -p|--purge) PURGE=YES;                         shift   ;;
        --three-way) THREEWAY=YES;                       shift   ;;
        --debconf-ok) DEBCONF_OK=YES;                    shift   ;;
+       -Z) selinux='-Z';                                shift   ;;
 	--)  shift ;                                   break   ;;
 	*) echo >&2 "Internal error!" ; exit 1 ;;
     esac
@@ -427,11 +438,19 @@
     fi
 fi
 
-
+# Follow dpkg-divert as though we are installed as part of $opt_package
+divert_line=$(dpkg-divert --listpackage "$dest_file")
+if [ -n "$divert_line" ]; then
+   # name of the package or 'LOCAL' for a local diversion
+   divert_package="$divert_line"
+
+   if [ "$divert_package" != "$opt_package" ]; then
+       dest_file=$(dpkg-divert --truename "$dest_file")
+   fi
+fi
 safe_dest_file=$(echo "$dest_file" | perl -nle 'print "\Q$_\E\n"')
 
 
-
 ######################################################################
 ########                                                     #########
 ########              Set Default Values                     #########
@@ -562,10 +581,10 @@
 if [ -e "$statedir/hashfile" ]; then
     if [ "X$VERBOSE" != "X" ]; then
 	echo >&2 "The hash file exists"
-	echo >&2 egrep "[[:space:]]${safe_dest_file}$" "$statedir/hashfile"
-	egrep "[[:space:]]${safe_dest_file}$" "$statedir/hashfile" >&2 || true
+	echo >&2 "grep -E" "[[:space:]]${safe_dest_file}$" "$statedir/hashfile"
+	grep -E "[[:space:]]${safe_dest_file}$" "$statedir/hashfile" >&2 || true
     fi
-    lastsum=$(egrep "[[:space:]]${safe_dest_file}$" "$statedir/hashfile" | \
+    lastsum=$(grep -E "[[:space:]]${safe_dest_file}$" "$statedir/hashfile" | \
                    awk '{print $1;}' )
 fi
 
@@ -770,7 +789,7 @@
 		    fi
 		done
 	    elif [ -f "$old_mdsum_file" ]; then
-		oldsum=$(egrep "^${destsum}" "$old_mdsum_file" || true)
+		oldsum=$(grep -E "^${destsum}" "$old_mdsum_file" || true)
 		if [ "X$oldsum" != "X" ]; then
 #                    Bingo
 		    if [ "X$force_conffold" = "X" ]; then
@@ -795,7 +814,7 @@
 #	   (like just one), the maintainer is also making a guess at
 #	   this point by supplying a historical md5sum default file.
 	    if [ "X$VERBOSE" != "X" ]; then
-		echo >&2 "Histotical md5sums did not match."
+		echo >&2 "Historical md5sums did not match."
 	    fi
 	    if [ -d "$old_mdsum_dir"  ]; then
 		if [ -e "${old_mdsum_dir}/default" ]; then
@@ -806,7 +825,7 @@
 		    do_replace_md5sum=1;
 		fi
 	    elif [ -f "$old_mdsum_file" ]; then
-		oldsum=$(egrep "[[:space:]]default$" "$old_mdsum_file" | \
+		oldsum=$(grep -E "[[:space:]]default$" "$old_mdsum_file" | \
 		    awk '{print $1;}')
 		if [ "X$oldsum" != "X" ]; then
 #                    Bingo
@@ -1084,6 +1103,8 @@
                 # </dev/tty). The easiest way do this from a shell is
                 # probably with /bin/ps.
                 if ps -o stat= --ppid $$ | grep -q '+'; then
+                    export UCF_CONFFILE_OLD="$dest_file"
+                    export UCF_CONFFILE_NEW="$new_file"
 		    bash >/dev/tty </dev/tty || true
                 elif [ -n "$DISPLAY" ]; then
                     x-terminal-emulator || true
