--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/share/automake-1.16/depcomp	2018-11-04 23:03:34.000000000 -0500
+++ /media/mks/armbi_root/usr/share/automake-1.16/depcomp	2022-03-18 09:09:08.000000000 -0400
@@ -3,7 +3,7 @@
 
 scriptversion=2018-03-07.03; # UTC
 
-# Copyright (C) 1999-2018 Free Software Foundation, Inc.
+# Copyright (C) 1999-2021 Free Software Foundation, Inc.
 
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
