--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/sbin/i2c-stub-from-dump	2018-12-04 14:37:48.000000000 -0500
+++ /media/mks/armbi_root/usr/sbin/i2c-stub-from-dump	2023-01-31 10:05:09.000000000 -0500
@@ -127,10 +127,10 @@
 	open(DUMP, $dump) || die "Can't open $dump: $!\n";
  OUTER_LOOP:
 	while (<DUMP>) {
-		if (m/^([0-9a-f]0) ?[:|](( [0-9a-fX]{2}){16})/i) {
+		if (m/^([0-9a-f]+0)[ :|]?((  ?[0-9a-fX]{2}){16})/i) {
 			# Byte dump
 			my $offset = hex($1);
-			my @values = split(/ /, $2);
+			my @values = split(/  ?/, $2);
 			shift(@values);
 			for (my $i = 0; $i < 16 && (my $val = shift(@values)); $i++) {
 				next if $val =~ m/X/;
