--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/share/doc/dialog/examples/editbox-utf8	2010-01-13 05:20:03.000000000 -0500
+++ /media/mks/armbi_root/usr/share/doc/dialog/examples/editbox-utf8	2020-11-25 19:03:58.000000000 -0500
@@ -1,5 +1,5 @@
 #!/bin/sh
-# $Id: editbox-utf8,v 1.9 2010/01/13 10:20:03 tom Exp $
+# $Id: editbox-utf8,v 1.10 2020/11/26 00:03:58 tom Exp $
 
 . ./setup-vars
 
@@ -30,6 +30,6 @@
 
 $DIALOG --title "EDIT BOX" \
 	--fixed-font "$@" --editbox $input 0 0 2>$output
-retval=$?
+returncode=$?
 
 . ./report-edit
