--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/bin/perlivp	2020-07-21 15:27:00.000000000 -0400
+++ /media/mks/armbi_root/usr/bin/perlivp	2023-11-25 15:59:54.000000000 -0500
@@ -1,8 +1,8 @@
 #!/usr/bin/perl
     eval 'exec /usr/bin/perl -S $0 ${1+"$@"}'
-        if $running_under_some_shell;
+        if 0; # ^ Run only under a shell
 
-# perlivp v5.28.1
+# perlivp v5.36.0
 
 BEGIN { pop @INC if $INC[-1] eq '.' }
 
@@ -67,7 +67,7 @@
 
 print "## Checking Perl version via variable '\$]'.\n" if $opt{'p'};
 
-my $ivp_VERSION = "5.028001";
+my $ivp_VERSION = "5.036000";
 
 
 $label = 'Perl version correct';
@@ -109,7 +109,7 @@
     $INC_total++;
 }
 
-$label = '@INC directoreis exist';
+$label = '@INC directories exist';
 if ($INC_total == $INC_there) {
     print "ok 3 $label\n";
     $pass__total++;
