--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/share/doc/dialog/examples/inputmenu2	2012-06-30 21:00:34.000000000 -0400
+++ /media/mks/armbi_root/usr/share/doc/dialog/examples/inputmenu2	2020-11-25 19:29:26.000000000 -0500
@@ -1,5 +1,5 @@
 #! /bin/sh
-# $Id: inputmenu2,v 1.9 2012/07/01 01:00:34 tom Exp $
+# $Id: inputmenu2,v 1.12 2020/11/26 00:29:26 tom Exp $
 #
 # "inputmenu1" with defaultitem, help-button.
 
@@ -19,7 +19,7 @@
 while test $returncode != 1 && test $returncode != 250
 do
 exec 3>&1
-value=`$DIALOG --clear --ok-label "Create" \
+returntext=`$DIALOG --clear --ok-label "Create" \
 	  --backtitle "$backtitle" \
 	  --help-button \
 	  --help-label "Script" \
@@ -65,8 +65,8 @@
 			$user" 10 40
 		;;
 	$DIALOG_EXTRA)
-		tag=`echo "$value" |sed -e 's/^RENAMED //' -e 's/:.*/:/'`
-		item=`echo "$value" |sed -e 's/^[^:]*:[ ]*//' -e 's/[ ]*$//'`
+		tag=`echo "$returntext" |sed -e 's/^RENAMED //' -e 's/:.*/:/'`
+		item=`echo "$returntext" |sed -e 's/^[^:]*:[ ]*//' -e 's/[ ]*$//'`
 
 		case "$tag" in
 		Username:)
@@ -88,8 +88,8 @@
 		test -n "$tag" && defaultitem="$tag"
 		;;
 
-	$DIALOG_ESC)
-                echo "ESC pressed."
+	*)
+                . ./report-button
                 break
                 ;;
 
