--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/bin/shasum.bundled	2020-07-21 15:27:00.000000000 -0400
+++ /media/mks/armbi_root/usr/bin/shasum.bundled	2023-11-25 15:59:54.000000000 -0500
@@ -1,14 +1,14 @@
 #!/usr/bin/perl
     eval 'exec /usr/bin/perl -S $0 ${1+"$@"}'
-	if $running_under_some_shell;
+	if 0; # ^ Run only under a shell
 #!perl
 
 	## shasum: filter for computing SHA digests (ref. sha1sum/md5sum)
 	##
-	## Copyright (C) 2003-2017 Mark Shelor, All Rights Reserved
+	## Copyright (C) 2003-2018 Mark Shelor, All Rights Reserved
 	##
-	## Version: 6.01
-	## Mon Dec 25 00:08:08 MST 2017
+	## Version: 6.02
+	## Fri Apr 20 16:25:30 MST 2018
 
 	## shasum SYNOPSIS adapted from GNU Coreutils sha1sum. Add
 	## "-a" option for algorithm selection,
@@ -100,7 +100,7 @@
 
 =head1 AUTHOR
 
-Copyright (c) 2003-2017 Mark Shelor <mshelor@cpan.org>.
+Copyright (C) 2003-2018 Mark Shelor <mshelor@cpan.org>.
 
 =head1 SEE ALSO
 
@@ -110,7 +110,7 @@
 
 END_OF_POD
 
-my $VERSION = "6.01";
+my $VERSION = "6.02";
 
 sub usage {
 	my($err, $msg) = @_;
