--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/lib/bridge-utils/bridge-utils.sh	2019-01-16 03:58:07.000000000 -0500
+++ /media/mks/armbi_root/usr/lib/bridge-utils/bridge-utils.sh	2023-01-25 16:11:52.000000000 -0500
@@ -8,7 +8,7 @@
     case $1 in
       all)
 	shift &&
-	set regex eth.\* em.\* 'p[0-9].*' noregex "$@"
+	set regex eth.\* em.\* en.\* 'p[0-9].*' noregex "$@"
 	;;
     esac
 
@@ -64,12 +64,12 @@
   # port is a vlan and the device exists?
   if [ "$port" != "$dev" ] && [ -e "/sys/class/net/$dev" ]
   then
-    if [ -f /proc/sys/net/ipv6/conf/$dev/disable_ipv6 ]
+    if [ "$BRIDGE_DISABLE_LINKLOCAL_IPV6_ALSO_PHYS" != "no" ] && [ -e "/proc/sys/net/ipv6/conf/$dev" ]
     then
-      echo 1 > /proc/sys/net/ipv6/conf/$dev/disable_ipv6
+      ip link set $dev addrgenmode none
     fi
     ip link set "$dev" up
-    ip link add link "$dev" name "$port" type vlan id "${port#*.}"
+    ip link add link "$dev" name "$port" type vlan id "$(echo $port|sed 's/[^.]*\.0*//g')"
   fi
 fi
 }
