--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/share/doc/dialog/examples/pause	2010-01-13 19:27:23.000000000 -0500
+++ /media/mks/armbi_root/usr/share/doc/dialog/examples/pause	2020-11-25 19:05:11.000000000 -0500
@@ -1,10 +1,10 @@
 #!/bin/sh
-# $Id: pause,v 1.5 2010/01/14 00:27:23 tom Exp $
+# $Id: pause,v 1.6 2020/11/26 00:05:11 tom Exp $
 
 . ./setup-vars
 
 $DIALOG --title "PAUSE" "$@" --pause "Hi, this is a pause widget" 20 70 10
 
-retval=$?
+returncode=$?
 
 . ./report-button
