--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/lib/mc/ext.d/archive.sh	2019-01-18 08:10:34.000000000 -0500
+++ /media/mks/armbi_root/usr/lib/mc/ext.d/archive.sh	2023-02-14 07:56:55.000000000 -0500
@@ -79,14 +79,18 @@
         ;;
 
     lha)
-        lha l "${MC_EXT_FILENAME}"
+        lha l "${MC_EXT_FILENAME}" 2>/dev/null || \
+            jlha l "${MC_EXT_FILENAME}" 2>/dev/null || \
+            lhasa l "${MC_EXT_FILENAME}" 2>/dev/null
         ;;
     arj)
         arj l "${MC_EXT_FILENAME}" 2>/dev/null || \
-            unarj l "${MC_EXT_FILENAME}"
+            unarj l "${MC_EXT_FILENAME}" 2>/dev/null || \
+            7za l "${MC_EXT_FILENAME}" 2>/dev/null
         ;;
     cab)
-        cabextract -l "${MC_EXT_FILENAME}"
+        cabextract -l "${MC_EXT_FILENAME}" 2> /dev/null || \
+            7za l "${MC_EXT_FILENAME}" 2>/dev/null
         ;;
     ha)
         ha lf "${MC_EXT_FILENAME}"
@@ -133,11 +137,18 @@
         arc l "${MC_EXT_FILENAME}"
         ;;
     zip)
-        unzip -v "${MC_EXT_FILENAME}"
+        unzip -v "${MC_EXT_FILENAME}" 2> /dev/null || \
+            7za l "${MC_EXT_FILENAME}" 2> /dev/null
+        ;;
+    zipx)
+        7za l "${MC_EXT_FILENAME}" 2> /dev/null
         ;;
     zoo)
         zoo l "${MC_EXT_FILENAME}"
         ;;
+    wim)
+        wimlib-imagex info "${MC_EXT_FILENAME}" 2> /dev/null
+        ;;
     *)
         ;;
     esac
