--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/sbin/installkernel	2019-01-21 16:12:11.000000000 -0500
+++ /media/mks/armbi_root/usr/sbin/installkernel	2023-07-28 19:46:35.000000000 -0400
@@ -42,7 +42,7 @@
   cat "$2" > "$dir/$1-$ver"
 
   # This section is for backwards compatibility only
-  if test -f "$dir/$1" ; then
+  if test -f "$dir/$1" || test -h "$dir/$1" ; then
     # The presence of "$dir/$1" is unusual in modern intallations, and
     # the results are mostly unused.  So only recreate them if they
     # already existed.
