--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/bin/bashbug	2019-04-18 00:12:36.000000000 -0400
+++ /media/mks/armbi_root/usr/bin/bashbug	2024-03-29 17:21:26.000000000 -0400
@@ -4,10 +4,10 @@
 #
 # The bug address depends on the release status of the shell.  Versions
 # with status `devel', `alpha', `beta', or `rc' mail bug reports to
-# chet@cwru.edu and, optionally, to bash-testers@cwru.edu.
+# chet.ramey@case.edu and, optionally, to bash-testers@cwru.edu.
 # Other versions send mail to bug-bash@gnu.org.
 #
-# Copyright (C) 1996-2004 Free Software Foundation, Inc.
+# Copyright (C) 1996-2021 Free Software Foundation, Inc.
 #
 #   This program is free software: you can redistribute it and/or modify
 #   it under the terms of the GNU General Public License as published by
@@ -29,9 +29,9 @@
 MACHINE="aarch64"
 OS="linux-gnu"
 CC="gcc"
-CFLAGS="-g -O2 -fdebug-prefix-map=/build/bash-gz1Plw/bash-5.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wall -Wno-parentheses -Wno-format-security"
-RELEASE="5.0"
-PATCHLEVEL="3"
+CFLAGS="-g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wall"
+RELEASE="5.2"
+PATCHLEVEL="15"
 RELSTATUS="release"
 MACHTYPE="aarch64-unknown-linux-gnu"
 
@@ -43,7 +43,7 @@
 
 #Securely create a temporary directory for the temporary files
 TEMPDIR=$TMPDIR/bbug.$$
-(umask 077 && mkdir $TEMPDIR) || {
+(umask 077 && mkdir "$TEMPDIR") || {
 	echo "$0: could not create temporary directory" >&2
 	exit 1
 }
@@ -102,7 +102,7 @@
 BASHTESTERS="bash-testers@cwru.edu"
 
 case "$RELSTATUS" in
-alpha*|beta*|devel*|rc*)	BUGBASH=chet@cwru.edu ;;
+alpha*|beta*|devel*|rc*)	BUGBASH=chet.ramey@case.edu ;;
 *)				BUGBASH=bug-bash@gnu.org ;;
 esac
 
@@ -132,6 +132,12 @@
 		DEFEDITOR=emacs
 	elif [ -x /usr/bin/xemacs ]; then
 		DEFEDITOR=xemacs
+	elif [ -x /usr/bin/vim; then
+		DEFEDITOR=vim
+	elif [ -x /usr/bin/gvim; then
+		DEFEDITOR=gvim
+	elif [ -x /usr/bin/nano ]; then
+		DEFEDITOR=nano
 	elif [ -x /usr/contrib/bin/jove ]; then
 		DEFEDITOR=jove
 	elif [ -x /usr/local/bin/jove ]; then
