--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/bin/dpkg-genchanges	2022-05-24 07:40:09.000000000 -0400
+++ /media/mks/armbi_root/usr/bin/dpkg-genchanges	2023-05-10 22:04:01.000000000 -0400
@@ -31,7 +31,7 @@
 use Dpkg::File;
 use Dpkg::Checksums;
 use Dpkg::ErrorHandling;
-use Dpkg::Build::Types;
+use Dpkg::BuildTypes;
 use Dpkg::BuildProfiles qw(get_build_profiles parse_build_profiles
                            evaluate_restriction_formula);
 use Dpkg::Arch qw(get_host_arch debarch_eq debarch_is debarch_list_parse);
@@ -44,6 +44,7 @@
 use Dpkg::Changelog::Parse;
 use Dpkg::Dist::Files;
 use Dpkg::Version;
+use Dpkg::Vendor qw(run_vendor_hook);
 
 textdomain('dpkg-dev');
 
@@ -128,7 +129,7 @@
 
 
 while (@ARGV) {
-    $_=shift(@ARGV);
+    $_ = shift @ARGV;
     if (m/^--build=(.*)$/) {
         set_build_type_from_options($1, $_);
     } elsif (m/^-b$/) {
@@ -144,9 +145,9 @@
     } elsif (m/^-g$/) {
 	set_build_type(BUILD_SOURCE | BUILD_ARCH_INDEP, $_);
     } elsif (m/^-s([iad])$/) {
-        $sourcestyle= $1;
+        $sourcestyle = $1;
     } elsif (m/^-q$/) {
-        $quiet= 1;
+        $quiet = 1;
     } elsif (m/^-c(.*)$/) {
 	$controlfile = $1;
     } elsif (m/^-l(.*)$/) {
@@ -212,17 +213,20 @@
 my $binaryversion = $changelog->{'Version'};
 
 $substvars->set_version_substvars($sourceversion, $binaryversion);
+$substvars->set_vendor_substvars();
 $substvars->set_arch_substvars();
 $substvars->load('debian/substvars') if -e 'debian/substvars' and not $substvars_loaded;
 
-if (defined($prev_changelog) and
+my $backport_version_regex = run_vendor_hook('backport-version-regex') // qr/^$/;
+my $is_backport = $changelog->{'Version'} =~ m/$backport_version_regex/;
+
+# Versions with backport markers have a lower version number by definition.
+if (! $is_backport && defined $prev_changelog &&
     version_compare_relation($changelog->{'Version'}, REL_LT,
                              $prev_changelog->{'Version'}))
 {
     warning(g_('the current version (%s) is earlier than the previous one (%s)'),
-	$changelog->{'Version'}, $prev_changelog->{'Version'})
-        # ~bpo and ~vola are backports and have lower version number by definition
-        unless $changelog->{'Version'} =~ /~(?:bpo|vola)/;
+            $changelog->{'Version'}, $prev_changelog->{'Version'});
 }
 
 # Scan control info of source package
@@ -267,7 +271,13 @@
     if (defined($prev_changelog)) {
         my $cur = Dpkg::Version->new($changelog->{'Version'});
         my $prev = Dpkg::Version->new($prev_changelog->{'Version'});
-        $include_tarball = ($cur->version() ne $prev->version()) ? 1 : 0;
+        if ($cur->version() ne $prev->version()) {
+            $include_tarball = 1;
+        } elsif ($changelog->{'Source'} ne $prev_changelog->{'Source'}) {
+            $include_tarball = 1;
+        } else {
+            $include_tarball = 0;
+        }
     } else {
         # No previous entry means first upload, tarball required
         $include_tarball = 1;
@@ -313,8 +323,10 @@
 
 foreach my $file ($dist->get_files()) {
     my $f = $file->{filename};
+    my $p = $file->{package};
+    my $a = $file->{arch};
 
-    if (defined $file->{package} && $file->{package_type} eq 'buildinfo') {
+    if (defined $p && $file->{package_type} eq 'buildinfo') {
         # We always distribute the .buildinfo file.
         $checksums->add_from_file("$uploadfilesdir/$f", key => $f);
         next;
@@ -323,17 +335,17 @@
     # If this is a source-only upload, ignore any other artifacts.
     next if build_has_none(BUILD_BINARY);
 
-    if (defined $file->{arch}) {
-        my $arch_all = debarch_eq('all', $file->{arch});
+    if (defined $a) {
+        my $arch_all = debarch_eq('all', $a);
 
         next if build_has_none(BUILD_ARCH_INDEP) and $arch_all;
         next if build_has_none(BUILD_ARCH_DEP) and not $arch_all;
 
-        push @archvalues, $file->{arch} if not $archadded{$file->{arch}}++;
+        push @archvalues, $a if not $archadded{$a}++;
     }
-    if (defined $file->{package} && $file->{package_type} =~ m/^u?deb$/) {
-        $p2f{$file->{package}} //= [];
-        push @{$p2f{$file->{package}}}, $file->{filename};
+    if (defined $p && $file->{package_type} =~ m/^u?deb$/) {
+        $p2f{$p} //= [];
+        push @{$p2f{$p}}, $f;
     }
 
     $checksums->add_from_file("$uploadfilesdir/$f", key => $f);
@@ -459,7 +471,7 @@
 
 $fields->{'Format'} = $substvars->get('Format');
 
-if (!defined($fields->{'Date'})) {
+if (length $fields->{'Date'} == 0) {
     setlocale(LC_TIME, 'C');
     $fields->{'Date'} = POSIX::strftime('%a, %d %b %Y %T %z', localtime);
     setlocale(LC_TIME, '');
