--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/share/doc/zsh-common/examples/Misc/c2z	2022-02-13 13:16:43.000000000 -0500
+++ /media/mks/armbi_root/usr/share/doc/zsh-common/examples/Misc/c2z	2023-02-04 19:37:45.000000000 -0500
@@ -10,7 +10,7 @@
 # uses the csh to parse its own dot-files, then processes csh output to
 # convert the csh settings to zsh.
 #
-# When run as a zsh fuction, c2z runs csh as if it were an interactive
+# When run as a zsh function, c2z runs csh as if it were an interactive
 # shell whenever the parent zsh is interactive.  When run as a shell
 # script, the -i switch can be used to force this behavior.
 #
