--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/lib/mc/extfs.d/iso9660	2019-01-18 08:10:34.000000000 -0500
+++ /media/mks/armbi_root/usr/lib/mc/extfs.d/iso9660	2023-02-14 07:56:55.000000000 -0500
@@ -43,7 +43,7 @@
     r=$?
     test $r -gt 0 && return $r
 
-    echo "$lsl" | /bin/grep "^[-d]" | \
+    echo "$lsl" | grep "^[-d]" | \
     while read attr ln usr gr sz dt1 dt2 dt3 nm ; do
         len=$((${#nm} - 1))
         name=$(printf -- "$nm" | cut -c2-$len)  # remove quotes
@@ -93,32 +93,40 @@
 
 # tested to comply with isoinfo 2.0's output
 test_iso () {
+    ISOINFO=$(which isoinfo 2>/dev/null)
+    if test -z "$ISOINFO"; then
+        echo "isoinfo not found" >&2
+        return 1
+    fi
+
     CHARSET=$(locale charmap 2>/dev/null)
     if test -z "$CHARSET"; then
-        CHARSET=$(locale 2>/dev/null | /bin/grep LC_CTYPE | sed -n -e 's/.*\.\(.*\)"$/\1/p')
+        CHARSET=$(locale 2>/dev/null | grep LC_CTYPE | sed -n -e 's/.*\.\(.*\)"$/\1/p')
     fi
     if test -n "$CHARSET"; then
         CHARSET=$(echo "$CHARSET" | tr '[A-Z]' '[a-z]' | sed -e 's/^iso-/iso/')
-        isoinfo -j $CHARSET -i /dev/null 2>&1 | /bin/grep "Iconv not yet supported\|Unknown charset" >/dev/null && CHARSET=
+        $ISOINFO -j $CHARSET -i /dev/null 2>&1 | grep "Iconv not yet supported\|Unknown charset" >/dev/null && CHARSET=
     fi
     if test -n "$CHARSET"; then
         JOLIET_OPT="-j $CHARSET -J"
     else
         JOLIET_OPT="-J"
     fi
-    ISOINFO="isoinfo -R"
 
-    ISOINFO_D_I="$(isoinfo -d -i "$1" 2>/dev/null)"
+    ISOINFO_D_I="$($ISOINFO -d -i "$1" 2>/dev/null)"
+    ISOINFO="$ISOINFO -R"
 
-    echo "$ISOINFO_D_I" | /bin/grep "UCS level 1\|NO Joliet" > /dev/null || ISOINFO="$ISOINFO $JOLIET_OPT"
+    echo "$ISOINFO_D_I" | grep "UCS level 1\|NO Joliet" > /dev/null || ISOINFO="$ISOINFO $JOLIET_OPT"
 
-    if [ $(echo "$ISOINFO_D_I" | /bin/grep "Joliet with UCS level 3 found" | wc -l) = 1 \
-        -a $(echo "$ISOINFO_D_I" | /bin/grep "NO Rock Ridge" | wc -l) = 1 ] ; then
+    if [ $(echo "$ISOINFO_D_I" | grep "Joliet with UCS level 3 found" | wc -l) = 1 \
+        -a $(echo "$ISOINFO_D_I" | grep "NO Rock Ridge" | wc -l) = 1 ] ; then
         SEMICOLON="YES"
     fi
 }
 
 mcisofs_list () {
+    local lsl r
+
     # left as a reminder to implement compressed image support =)
     case "$1" in
       *.lz)  MYCAT="lzip -dc";;
@@ -133,7 +141,11 @@
       *)     MYCAT="cat";;
     esac
 
-    $ISOINFO -l -i "$1" 2>/dev/null | awk -v SEMICOLON=$SEMICOLON '
+    lsl=$($ISOINFO -l -i "$1"  2>/dev/null)
+    r=$?
+    test $r -gt 0 && return $r
+
+    echo "$lsl" | awk -v SEMICOLON=$SEMICOLON '
 BEGIN {
   dir="";
   # Pattern to match 8 first fields.
@@ -183,8 +195,8 @@
 case "$cmd" in
   list)
     xorriso_list "$@" || {
-        test_iso "$@";
-        mcisofs_list "$@";
+        test_iso "$@" || exit 1
+        mcisofs_list "$@" || exit 1
     }
     exit 0
     ;;
@@ -214,8 +226,8 @@
     ;;
   copyout)
     xorriso_copyout "$@" || {
-        test_iso "$@";
-        mcisofs_copyout "$@";
+        test_iso "$@" || exit 1
+        mcisofs_copyout "$@" || exit 1
     }
     exit 0
     ;;
