--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/share/doc/python3-libgpiod/examples/gpioinfo.py	2018-11-09 03:31:22.000000000 -0500
+++ /media/mks/armbi_root/usr/share/doc/python3-libgpiod/examples/gpioinfo.py	2021-03-22 08:25:56.000000000 -0400
@@ -11,21 +11,22 @@
 
 import gpiod
 
-for chip in gpiod.ChipIter():
-    print('{} - {} lines:'.format(chip.name(), chip.num_lines()))
+if __name__ == '__main__':
+    for chip in gpiod.ChipIter():
+        print('{} - {} lines:'.format(chip.name(), chip.num_lines()))
 
-    for line in gpiod.LineIter(chip):
-        offset = line.offset()
-        name = line.name()
-        consumer = line.consumer()
-        direction = line.direction()
-        active_state = line.active_state()
+        for line in gpiod.LineIter(chip):
+            offset = line.offset()
+            name = line.name()
+            consumer = line.consumer()
+            direction = line.direction()
+            active_state = line.active_state()
 
-        print('\tline {:>3}: {:>18} {:>12} {:>8} {:>10}'.format(
-                offset,
-                'unnamed' if name is None else name,
-                'unused' if consumer is None else consumer,
-                'input' if direction == gpiod.Line.DIRECTION_INPUT else 'output',
-                'active-low' if active_state == gpiod.Line.ACTIVE_LOW else 'active-high'))
+            print('\tline {:>3}: {:>18} {:>12} {:>8} {:>10}'.format(
+                    offset,
+                    'unnamed' if name is None else name,
+                    'unused' if consumer is None else consumer,
+                    'input' if direction == gpiod.Line.DIRECTION_INPUT else 'output',
+                    'active-low' if active_state == gpiod.Line.ACTIVE_LOW else 'active-high'))
 
-    chip.close()
+        chip.close()
