--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/bin/perlthanks	2020-07-21 15:27:00.000000000 -0400
+++ /media/mks/armbi_root/usr/bin/perlthanks	2023-11-25 15:59:54.000000000 -0500
@@ -1,10 +1,10 @@
 #!/usr/bin/perl
     eval 'exec /usr/bin/perl -S $0 ${1+"$@"}'
-	if $running_under_some_shell;
+	if 0; # ^ Run only under a shell
 
-my $config_tag1 = '5.28.1 - Tue Jul 21 19:27:00 UTC 2020';
+my $config_tag1 = '5.36.0 - Sat Nov 25 20:59:54 UTC 2023';
 
-my $patchlevel_date = 1595359601;
+my $patchlevel_date = 1700945994;
 my @patches = Config::local_patches();
 my $patch_tags = join "", map /(\S+)/ ? "+$1 " : (), @patches;
 
@@ -34,14 +34,14 @@
     $::HaveWrap = ($@ eq "");
 };
 
-our $VERSION = "1.41";
+our $VERSION = "1.42";
 
 #TODO:
 #       make sure failure (transmission-wise) of Mail::Send is accounted for.
 #       (This may work now. Unsure of the original author's issue -JESSE 2008-06-08)
 #       - Test -b option
 
-my( $file, $usefile, $cc, $address, $bugaddress, $testaddress, $thanksaddress,
+my( $file, $usefile, $cc, $address, $thanksaddress,
     $filename, $messageid, $domain, $subject, $from, $verbose, $ed, $outfile,
     $fh, $me, $body, $andcc, %REP, $ok, $thanks, $progname,
     $Is_MSWin32, $Is_Linux, $Is_VMS, $Is_OpenBSD,
@@ -70,9 +70,7 @@
 Query();
 Edit() unless $usefile || ($ok and not $opt{n});
 NowWhat();
-if ($outfile) {
-    save_message_to_disk($outfile);
-} else {
+if ($address) {
     Send();
     if ($thanks) {
 	print "\nThank you for taking the time to send a thank-you message!\n\n";
@@ -86,12 +84,13 @@
 
 	paraprint <<EOF
 Please note that mailing lists are moderated, your message may take a while to
-show up. If you do not receive an automated response acknowledging your message
-within a few hours (check your SPAM folder and outgoing mail) please consider
-sending an email directly from your mail client to perlbug\@perl.org.
+show up. Please consider submitting your report directly to the issue tracker
+at https://github.com/Perl/perl5/issues
 EOF
     }
 
+} else {
+    save_message_to_disk($outfile);
 }
 
 exit;
@@ -149,17 +148,11 @@
     $Is_Linux = lc($^O) eq 'linux';
     $Is_OpenBSD = lc($^O) eq 'openbsd';
 
-    # perlbug address
-    $bugaddress = 'perlbug@perl.org';
-
-    # Test address
-    $testaddress = 'perlbug-test@perl.org';
-
     # Thanks address
     $thanksaddress = 'perl-thanks@perl.org';
 
     # Defaults if getopts fails.
-    $address = (basename ($0) =~ /^perlthanks/i) ? $thanksaddress : $bugaddress;
+    $outfile = (basename($0) =~ /^perlthanks/i) ? "perlthanks.rep" : "perlbug.rep";
     $cc = $::Config{'perladmin'} || $::Config{'cf_email'} || $::Config{'cf_by'} || '';
 
     HELP_MESSAGE() unless getopts("Adhva:s:b:f:F:r:e:SCc:to:n:T:p:", \%opt);
@@ -180,8 +173,7 @@
     
     $progname = $thanks ? 'perlthanks' : 'perlbug';
     # Target address
-    $address = $opt{a} || ($opt{t} ? $testaddress
-			    : $thanks ? $thanksaddress : $bugaddress);
+    $address = $opt{a} || ($thanks ? $thanksaddress : "");
 
     # Users address, used in message and in From and Reply-To headers
     $from = $opt{r} || "";
@@ -214,7 +206,7 @@
     }
 
     # File to output to
-    $outfile = $opt{F} || "";
+    $outfile = $opt{F} || "$progname.rep";
 
     # Body of report
     $body = $opt{b} || "";
@@ -309,21 +301,22 @@
 This program provides an easy way to send a thank-you message back to the
 authors and maintainers of perl.
 
-If you wish to submit a bug report, please run it without the -T flag
+If you wish to generate a bug report, please run it without the -T flag
 (or run the program perlbug rather than perlthanks)
 EOF
 	} else {
 	    paraprint <<"EOF";
-This program provides an easy way to create a message reporting a
-bug in the core perl distribution (along with tests or patches)
-to the volunteers who maintain perl at $address.  To send a thank-you
+This program provides an easy way to generate a bug report for the core
+perl distribution (along with tests or patches).  To send a thank-you
 note to $thanksaddress instead of a bug report, please run 'perlthanks'.
 
-Please do not use $0 to send test messages, test whether perl
-works, or to report bugs in perl modules from CPAN.
+The GitHub issue tracker at https://github.com/Perl/perl5/issues is the
+best place to submit your report so it can be tracked and resolved.
+
+Please do not use $0 to report bugs in perl modules from CPAN.
 
 Suggestions for how to find help using Perl can be found at
-http://perldoc.perl.org/perlcommunity.html
+https://perldoc.perl.org/perlcommunity.html
 EOF
 	}
     }
@@ -336,7 +329,7 @@
 
     unless ($subject) {
 	    print 
-"First of all, please provide a subject for the message.\n";
+"First of all, please provide a subject for the report.\n";
 	if ( not $thanks)  {
 	    paraprint <<EOF;
 This should be a concise description of your bug or problem
@@ -414,7 +407,7 @@
     }
 
     # Prompt for administrator address, unless an override was given
-    if( !$opt{C} and !$opt{c} ) {
+    if( $address and !$opt{C} and !$opt{c} ) {
 	my $description =  <<EOF;
 $0 can send a copy of this report to your local perl
 administrator.  If the address below is wrong, please correct it,
@@ -504,7 +497,7 @@
 		paraprint <<EOF;
 $entry is not a "core" Perl module. Please check that you entered
 its name correctly. If it is correct, quit this program, try searching
-for $entry on http://rt.cpan.org, and report your issue there.
+for $entry on https://rt.cpan.org, and report your issue there.
 EOF
 
             $entry = '';
@@ -644,11 +637,6 @@
     module=$report_about_module
 EFF
     }
-    if ($opt{A}) {
-	print OUT <<EFF;
-    ack=no
-EFF
-    }
     print OUT <<EFF;
 ---
 EFF
@@ -737,8 +725,7 @@
                 next;
             } else {
                 paraprint <<EOF;
-You may want to save your report to a file, so you can edit and
-mail it later.
+You can edit your report after saving it to a file.
 EOF
                 return;
             }
@@ -768,7 +755,7 @@
 
 sub Cancel {
     1 while unlink($filename);  # remove all versions under VMS
-    print "\nQuitting without sending your message.\n";
+    print "\nQuitting without generating a report.\n";
     exit(0);
 }
 
@@ -776,23 +763,24 @@
     # Report is done, prompt for further action
     if( !$opt{S} ) {
 	while(1) {
+	    my $send_to = $address || 'the Perl developers';
 	    my $menu = <<EOF;
 
 
-You have finished composing your message. At this point, you have 
+You have finished composing your report. At this point, you have 
 a few options. You can:
 
-    * [Se]nd the message to $address$andcc, 
-    * [D]isplay the message on the screen,
-    * [R]e-edit the message
-    * Display or change the message's [su]bject
-    * Save the message to a [f]ile to mail at another time
-    * [Q]uit without sending a message
+    * Save the report to a [f]ile
+    * [Se]nd the report to $send_to$andcc
+    * [D]isplay the report on the screen
+    * [R]e-edit the report
+    * Display or change the report's [su]bject
+    * [Q]uit without generating the report
 
 EOF
       retry:
         print $menu;
-	    my $action =  _prompt('', "Action (Send/Display/Edit/Subject/Save to File)",
+	    my $action =  _prompt('', "Action (Save/Send/Display/Edit/Subject/Quit)",
 	        $opt{t} ? 'q' : '');
         print "\n";
 	    if ($action =~ /^(f|sa)/i) { # <F>ile/<Sa>ve
@@ -814,12 +802,20 @@
 		}
 	    } elsif ($action =~ /^se/i) { # <S>end
 		# Send the message
-		my $reply =  _prompt( "Are you certain you want to send this message?", 'Please type "yes" if you are','no');
+		if (not $thanks) {
+		    print <<EOF
+To ensure your issue can be best tracked and resolved,
+you should submit it to the GitHub issue tracker at
+https://github.com/Perl/perl5/issues
+EOF
+		}
+		my $reply =  _prompt( "Are you certain you want to send this report to $send_to$andcc?", 'Please type "yes" if you are','no');
 		if ($reply =~ /^yes$/) {
+		    $address ||= 'perl5-porters@perl.org';
 		    last;
 		} else {
 		    paraprint <<EOF;
-You didn't type "yes", so your message has not yet been sent.
+You didn't type "yes", so your report has not been sent.
 EOF
 		}
 	    } elsif ($action =~ /^[er]/i) { # <E>dit, <R>e-edit
@@ -850,14 +846,9 @@
 }
 
 sub SaveMessage {
-    my $file_save = $outfile || "$progname.rep";
-    my $file = _prompt( '', "Name of file to save message in", $file_save );
+    my $file = _prompt( '', "Name of file to save report in", $outfile );
     save_message_to_disk($file) || return undef;
-    print "\n";
-    paraprint <<EOF;
-A copy of your message has been saved in '$file' for you to
-send to '$address' with your normal mail client.
-EOF
+    return 1;
 }
 
 sub Send {
@@ -893,7 +884,7 @@
 sub Help {
     print <<EOF;
 
-This program is designed to help you generate and send bug reports
+This program is designed to help you generate bug reports
 (and thank-you notes) about perl5 and the modules which ship with it.
 
 In most cases, you can just run "$0" interactively from a command
@@ -904,33 +895,33 @@
 $0  [-v] [-a address] [-s subject] [-b body | -f inpufile ] [ -F outputfile ]
     [-r returnaddress] [-e editor] [-c adminaddress | -C] [-S] [-t] [-h]
     [-p patchfile ]
-$0  [-v] [-r returnaddress] [-A] [-ok | -okay | -nok | -nokay]
+$0  [-v] [-r returnaddress] [-ok | -okay | -nok | -nokay]
 
 
 Options:
 
   -v    Include Verbose configuration data in the report
   -f    File containing the body of the report. Use this to
-        quickly send a prepared message.
+        quickly send a prepared report.
   -p    File containing a patch or other text attachment. Separate
         multiple files with commas.
-  -F    File to output the resulting mail message to, instead of mailing.
-  -S    Send without asking for confirmation.
-  -a    Address to send the report to. Defaults to '$address'.
+  -F    File to output the resulting report to. Defaults to
+        '$outfile'.
+  -S    Save or send the report without asking for confirmation.
+  -a    Send the report to this address, instead of saving to a file.
   -c    Address to send copy of report to. Defaults to '$cc'.
   -C    Don't send copy to administrator.
-  -s    Subject to include with the message. You will be prompted
+  -s    Subject to include with the report. You will be prompted
         if you don't supply one on the command line.
   -b    Body of the report. If not included on the command line, or
-        in a file with -f, you will get a chance to edit the message.
+        in a file with -f, you will get a chance to edit the report.
   -r    Your return address. The program will ask you to confirm
         this if you don't give it here.
   -e    Editor to use.
-  -t    Test mode. The target address defaults to '$testaddress'.
+  -t    Test mode.
   -T    Thank-you mode. The target address defaults to '$thanksaddress'.
   -d    Data mode.  This prints out your configuration data, without mailing
         anything. You can use this with -v to get more complete data.
-  -A    Don't send a bug received acknowledgement to the return address.
   -ok   Report successful build on this system to perl porters
         (use alone or with -v). Only use -ok if *everything* was ok:
         if there were *any* problems at all, use -nok.
@@ -997,7 +988,7 @@
 }
 
 sub _message_headers {
-    my %headers = ( To => $address, Subject => $subject );
+    my %headers = ( To => $address || 'perl5-porters@perl.org', Subject => $subject );
     $headers{'Cc'}         = $cc        if ($cc);
     $headers{'Message-Id'} = $messageid if ($messageid);
     $headers{'Reply-To'}   = $from      if ($from);
@@ -1082,7 +1073,7 @@
 
         print OUTFILE build_complete_message();
         close(OUTFILE) or do { warn  "Error closing $file: $!"; return undef };
-	    print "\nMessage saved.\n";
+	    print "\nReport saved to '$file'. Please submit it to https://github.com/Perl/perl5/issues\n";
         return 1;
 }
 
@@ -1157,9 +1148,9 @@
         paraprint(<<"EOF"), die "\n";
 $message_start
 Because of this, there's no easy way to automatically send your
-message.
+report.
 
-A copy of your message has been saved in '$filename' for you to
+A copy of your report has been saved in '$filename' for you to
 send to '$address' with your normal mail client.
 EOF
     }
@@ -1219,17 +1210,17 @@
 S<[ B<-b> I<body> | B<-f> I<inputfile> ]> S<[ B<-F> I<outputfile> ]>
 S<[ B<-r> I<returnaddress> ]>
 S<[ B<-e> I<editor> ]> S<[ B<-c> I<adminaddress> | B<-C> ]>
-S<[ B<-S> ]> S<[ B<-t> ]>  S<[ B<-d> ]>  S<[ B<-A> ]>  S<[ B<-h> ]> S<[ B<-T> ]>
+S<[ B<-S> ]> S<[ B<-t> ]>  S<[ B<-d> ]>  S<[ B<-h> ]> S<[ B<-T> ]>
 
 B<perlbug> S<[ B<-v> ]> S<[ B<-r> I<returnaddress> ]>
- S<[ B<-A> ]> S<[ B<-ok> | B<-okay> | B<-nok> | B<-nokay> ]>
+ S<[ B<-ok> | B<-okay> | B<-nok> | B<-nokay> ]>
 
 B<perlthanks>
 
 =head1 DESCRIPTION
 
 
-This program is designed to help you generate and send bug reports
+This program is designed to help you generate bug reports
 (and thank-you notes) about perl5 and the modules which ship with it.
 
 In most cases, you can just run it interactively from a command
@@ -1241,17 +1232,16 @@
 documentation that came with that distribution to determine the
 correct place to report bugs.
 
-If you are unable to send your report using B<perlbug> (most likely
-because your system doesn't have a way to send mail that perlbug
-recognizes), you may be able to use this tool to compose your report
-and save it to a file which you can then send to B<perlbug@perl.org>
-using your regular mail client.
+Bug reports should be submitted to the GitHub issue tracker at
+L<https://github.com/Perl/perl5/issues>. The B<perlbug@perl.org>
+address no longer automatically opens tickets. You can use this tool
+to compose your report and save it to a file which you can then submit
+to the issue tracker.
 
 In extreme cases, B<perlbug> may not work well enough on your system
 to guide you through composing a bug report. In those cases, you
-may be able to use B<perlbug -d> to get system configuration
-information to include in a manually composed bug report to
-B<perlbug@perl.org>.
+may be able to use B<perlbug -d> or B<perl -V> to get system
+configuration information to include in your issue report.
 
 
 When reporting a bug, please run through this checklist:
@@ -1264,7 +1254,7 @@
 
 =item Are you running the latest released version of perl?
 
-Look at http://www.perl.org/ to find out.  If you are not using the
+Look at L<http://www.perl.org/> to find out.  If you are not using the
 latest released version, please try to replicate your bug on the
 latest stable release.
 
@@ -1274,9 +1264,6 @@
 volunteers who build and maintain Perl than reports about bugs in
 the current release.
 
-This tool isn't appropriate for reporting bugs in any version
-prior to Perl 5.0.
-
 =item Are you sure what you have is a bug?
 
 A significant number of the bug reports we get turn out to be
@@ -1325,7 +1312,7 @@
 
 If you get a core dump (or equivalent), you may use a debugger
 (B<dbx>, B<gdb>, etc) to produce a stack trace to include in the bug
-report.  
+report.
 
 NOTE: unless your Perl has been compiled with debug info
 (often B<-g>), the stack trace is likely to be somewhat hard to use
@@ -1344,9 +1331,11 @@
 
 If so, that's great news; bug reports with patches are likely to
 receive significantly more attention and interest than those without
-patches.  Please attach your patch to the report using the C<-p> option.
-When sending a patch, create it using C<git format-patch> if possible,
-though a unified diff created with C<diff -pu> will do nearly as well.
+patches.  Please submit your patch via the GitHub Pull Request workflow
+as described in B<perldoc> L<perlhack>.  You may also send patches to
+B<perl5-porters@perl.org>.  When sending a patch, create it using
+C<git format-patch> if possible, though a unified diff created with
+C<diff -pu> will do nearly as well.
 
 Your patch may be returned with requests for changes, or requests for more
 detailed explanations about your fix.
@@ -1361,21 +1350,6 @@
 really does work (C<make test>, if the thing you're patching is covered
 by Perl's test suite).
 
-=item Can you use C<perlbug> to submit the report?
-
-B<perlbug> will, amongst other things, ensure your report includes
-crucial information about your version of perl.  If C<perlbug> is
-unable to mail your report after you have typed it in, you may have
-to compose the message yourself, add the output produced by C<perlbug
--d> and email it to B<perlbug@perl.org>.  If, for some reason, you
-cannot run C<perlbug> at all on your system, be sure to include the
-entire output produced by running C<perl -V> (note the uppercase V).
-
-Whether you use C<perlbug> or send the email manually, please make
-your Subject line informative.  "a bug" is not informative.  Neither
-is "perl crashes" nor is "HELP!!!".  These don't help.  A compact
-description of what's wrong is fine.
-
 =item Can you use C<perlbug> to submit a thank-you note?
 
 Yes, you can do this by either using the C<-T> option, or by invoking
@@ -1384,6 +1358,10 @@
 
 =back
 
+Please make your issue title informative.  "a bug" is not informative.
+Neither is "perl crashes" nor is "HELP!!!".  These don't help.  A compact
+description of what's wrong is fine.
+
 Having done your bit, please be prepared to wait, to be told the
 bug is in your code, or possibly to get no reply at all.  The
 volunteers who maintain Perl are busy folks, so if your problem is
@@ -1392,14 +1370,15 @@
 reply.
 
 If it is important to you that your bug be fixed, do monitor the
-perl5-porters@perl.org mailing list (mailing lists are moderated, your
-message may take a while to show up) and the commit logs to development
+issue tracker (you will be subscribed to notifications for issues you
+submit or comment on) and the commit logs to development
 versions of Perl, and encourage the maintainers with kind words or
 offers of frosty beverages.  (Please do be kind to the maintainers.
 Harassing or flaming them is likely to have the opposite effect of the
 one you want.)
 
-Feel free to update the ticket about your bug on http://rt.perl.org
+Feel free to update the ticket about your bug on
+L<https://github.com/Perl/perl5/issues>
 if a new version of Perl is released and your bug is still present.
 
 =head1 OPTIONS
@@ -1408,34 +1387,28 @@
 
 =item B<-a>
 
-Address to send the report to.  Defaults to B<perlbug@perl.org>.
-
-=item B<-A>
-
-Don't send a bug received acknowledgement to the reply address.
-Generally it is only a sensible to use this option if you are a
-perl maintainer actively watching perl porters for your message to
-arrive.
+Address to send the report to instead of saving to a file.
 
 =item B<-b>
 
 Body of the report.  If not included on the command line, or
-in a file with B<-f>, you will get a chance to edit the message.
+in a file with B<-f>, you will get a chance to edit the report.
 
 =item B<-C>
 
-Don't send copy to administrator.
+Don't send copy to administrator when sending report by mail.
 
 =item B<-c>
 
-Address to send copy of report to.  Defaults to the address of the
+Address to send copy of report to when sending report by mail.
+Defaults to the address of the
 local perl administrator (recorded when perl was built).
 
 =item B<-d>
 
 Data mode (the default if you redirect or pipe output).  This prints out
-your configuration data, without mailing anything.  You can use this
-with B<-v> to get more complete data.
+your configuration data, without saving or mailing anything.  You can use
+this with B<-v> to get more complete data.
 
 =item B<-e>
 
@@ -1444,13 +1417,11 @@
 =item B<-f>
 
 File containing the body of the report.  Use this to quickly send a
-prepared message.
+prepared report.
 
 =item B<-F>
 
-File to output the results to instead of sending as an email. Useful
-particularly when running perlbug on a machine with no direct internet
-connection.
+File to output the results to.  Defaults to B<perlbug.rep>.
 
 =item B<-h>
 
@@ -1496,17 +1467,16 @@
 
 =item B<-S>
 
-Send without asking for confirmation.
+Save or send the report without asking for confirmation.
 
 =item B<-s>
 
-Subject to include with the message.  You will be prompted if you don't
+Subject to include with the report.  You will be prompted if you don't
 supply one on the command line.
 
 =item B<-t>
 
-Test mode.  The target address defaults to B<perlbug-test@perl.org>.
-Also makes it possible to command perlbug from a pipe or file, for
+Test mode.  Makes it possible to command perlbug from a pipe or file, for
 testing purposes.
 
 =item B<-T>
