--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/bin/sensible-editor	2018-03-12 06:17:53.000000000 -0400
+++ /media/mks/armbi_root/usr/bin/sensible-editor	2023-01-14 11:28:42.000000000 -0500
@@ -1,44 +1,57 @@
 #!/bin/sh
-
+# Copyright 2007 Jari Aalto; Released under GNU GPL v2 or any later version
+# Copyright 201
 ret="$?"
 
 # Prevent recursive loops, where these values are set to this script
-p="$(which sensible-editor)"
-[ "$(which $EDITOR || true)" = "$p" ] && EDITOR=
-[ "$(which $VISUAL || true)" = "$p" ] && VISUAL=
-[ "$(which $SELECTED_EDITOR || true)" = "$p" ] && SELECTED_EDITOR=
+p="$(command -v "$0")"
+[ -n "$EDITOR" ] && [ "$(command -v "$EDITOR" || true)" = "$p" ] && EDITOR=
+[ -n "$VISUAL" ] && [ "$(command -v "$VISUAL" || true)" = "$p" ] && VISUAL=
+[ -n "$SELECTED_EDITOR" ] && [ "$(command -v "$SELECTED_EDITOR" || true)" = "$p" ] && SELECTED_EDITOR=
+
+IsError()
+{
+	# Operating system command not found
+	[ "$1" -ne 126 ] && [ $1 -ne 127 ]
+}
+
+Run()
+{
+	"$@"
+	ret=$?
+	IsError "$ret"
+}
+
+# workarround #991982
+nano ()
+{
+    if [ -z "$TERM" ]; then
+	return 126
+    else
+	command nano "$@"
+    fi
+}
 
 if [ -n "$VISUAL" ]; then
-	${VISUAL} "$@"
-	ret="$?"
-	if [ "$ret" -ne 126 ] && [ "$ret" -ne 127 ]; then
-		exit "$ret"
-	fi
+	Run "$VISUAL" "$@" && exit "$ret"
 fi
 
-if [ -r ~/.selected_editor ]; then
+# fix #987675
+if [ -n "$HOME" ]; then
+    if [ -r ~/.selected_editor ]; then
 	. ~/.selected_editor 2>/dev/null || true
-elif [ -z "$EDITOR" ] && [ -z "$SELECTED_EDITOR" ] && [ -t 0 ]; then
+    elif [ -z "$EDITOR" ] && [ -z "$SELECTED_EDITOR" ] && [ -t 0 ]; then
 	select-editor && . ~/.selected_editor 2>/dev/null || true
+    fi
 fi
 
-${EDITOR:-${SELECTED_EDITOR:-editor}} "$@"
-ret="$?"
-if [ "$ret" -eq 126 ] || [ "$ret" -eq 127 ]; then
-	nano "$@"
-	ret="$?"
-	if [ "$ret" -eq 126 ] || [ "$ret" -eq 127 ]; then
-		nano-tiny "$@"
-		ret="$?"
-		if [ "$ret" -eq 126 ] || [ "$ret" -eq 127 ]; then
-			vi "$@"
-			ret="$?"
-			if [ "$ret" -eq 126 ] || [ "$ret" -eq 127 ]; then
-				echo "Couldn't find an editor!" 1>&2
-				echo "Set the \$EDITOR environment variable to your desired editor." 1>&2
-				exit 1
-			fi
-		fi
-	fi
-fi
+Run ${EDITOR:-${SELECTED_EDITOR:-editor}} "$@" ||
+Run nano "$@" ||
+Run nano-tiny "$@" ||
+Run vi "$@" ||
+{
+	echo "Couldn't find an editor!" 1>&2
+	echo "Set the \$EDITOR environment variable to your desired editor." 1>&2
+	exit 1;
+}
 exit "$ret"
