--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/usr/bin/bzexe	2019-06-24 16:16:40.000000000 -0400
+++ /media/mks/armbi_root/usr/bin/bzexe	2021-11-27 07:25:05.000000000 -0500
@@ -120,8 +120,8 @@
 set -C
 umask=`umask`
 umask 77
-tmpfile=`tempfile -p gztmp -d /tmp` || exit 1
-if tail +$skip "$0" | /bin/bzip2 -cd >> $tmpfile; then
+tmpfile=`mktemp ${TMPDIR:-/tmp}/bzexe.XXXXXXXXXX` || exit 1
+if tail +$skip "$0" | /bin/bzip2 -cd >> "$tmpfile"; then
   umask $umask
   /bin/chmod 700 $tmpfile
   prog="`echo $0 | /bin/sed 's|^.*/||'`"
