--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/adduser.postrm	2018-09-15 15:12:39.000000000 -0400
+++ /media/mks/armbi_root/var/lib/dpkg/info/adduser.postrm	2023-05-25 11:54:35.000000000 -0400
@@ -1,14 +1,14 @@
 #! /bin/sh
-set -e
 
-if [ "$1" = "purge" ]; then
-    rm -f /etc/adduser.conf /etc/adduser.conf.dpkg-save
-fi
-
-# Automatically added by dh_installdebconf/11.4
-if [ "$1" = purge ] && [ -e /usr/share/debconf/confmodule ]; then
-	. /usr/share/debconf/confmodule
-	db_purge
-fi
-# End automatically added section
+
+set -eu
+
+case $1 in
+  purge)
+    rm -fv /etc/adduser.conf /etc/adduser.conf.dpkg-save /etc/adduser.conf.update-old
+    ;;
+esac
+
+set +eu
+
 
