--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/python3-numpy.postinst	2019-03-02 11:30:12.000000000 -0500
+++ /media/mks/armbi_root/var/lib/dpkg/info/python3-numpy.postinst	2023-02-08 22:54:23.000000000 -0500
@@ -1,12 +1,12 @@
 #!/bin/sh
 set -e
 
-# Automatically added by dh_python3:
-if which py3compile >/dev/null 2>&1; then
-	py3compile -p python3-numpy 
+# Automatically added by dh_python3
+if command -v py3compile >/dev/null 2>&1; then
+	py3compile -p python3-numpy:arm64 
 fi
-if which pypy3compile >/dev/null 2>&1; then
-	pypy3compile -p python3-numpy  || true
+if command -v pypy3compile >/dev/null 2>&1; then
+	pypy3compile -p python3-numpy:arm64  || true
 fi
 
 # End automatically added section
