--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/man-db.prerm	2019-02-10 07:14:20.000000000 -0500
+++ /media/mks/armbi_root/var/lib/dpkg/info/man-db.prerm	2023-03-12 18:23:59.000000000 -0400
@@ -1,7 +1,7 @@
 #!/bin/sh
 set -e
-# Automatically added by dh_systemd_start/12
-if [ -d /run/systemd/system ]; then
+# Automatically added by dh_installsystemd/13.11.4
+if [ -z "${DPKG_ROOT:-}" ] && [ "$1" = remove ] && [ -d /run/systemd/system ] ; then
 	deb-systemd-invoke stop 'man-db.service' 'man-db.timer' >/dev/null || true
 fi
 # End automatically added section
