--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/procps.prerm	2018-05-31 05:42:46.000000000 -0400
+++ /media/mks/armbi_root/var/lib/dpkg/info/procps.prerm	2022-12-19 01:06:38.000000000 -0500
@@ -1,15 +1,6 @@
 #!/bin/sh
 set -e
-
-case "$1" in
-  remove|deconfigure)
-	update-alternatives --remove w /usr/bin/w.procps
-    ;;
-  upgrade|failed-upgrade)
-    ;;
-esac
-
-
-
-exit 0
-
+# Automatically added by dh_installdeb/13.11.3
+dpkg-maintscript-helper rm_conffile /etc/sysctl.d/protect-links.conf 2:3.3.16-4\~ procps -- "$@"
+dpkg-maintscript-helper mv_conffile /usr/lib/sysctl.d/protect-links.conf /usr/lib/sysctl.d/99-protect-links.conf 2:3.3.17-6\~ procps -- "$@"
+# End automatically added section
