--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/libpam-runtime.postinst	2019-02-14 02:08:47.000000000 -0500
+++ /media/mks/armbi_root/var/lib/dpkg/info/libpam-runtime.postinst	2023-09-21 16:55:12.000000000 -0400
@@ -8,7 +8,7 @@
 	sed -n -e'1,/# here are the per-package modules (the "Primary" block)/p;
 	          /# here.s the fallback if no module succeeds/,/# and here are more per-package modules (the "Additional" block)/p;
 	          /# end of pam-auth-update config/,$p' \
-	    /etc/pam.d/"$configfile" | md5sum | awk '{ print $1 }'
+	    "$DPKG_ROOT"/etc/pam.d/"$configfile" | md5sum | awk '{ print $1 }'
 }
 
 # If the user has removed the config file, respect this sign of dementia
@@ -20,26 +20,22 @@
 	for configfile in common-auth common-account common-session  \
 	    common-password
 	do
-		if [ -f /etc/pam.d/$configfile ] && \
+		if [ -f "$DPKG_ROOT"/etc/pam.d/$configfile ] && \
 		    ! fgrep -q $(calculate_md5sum $configfile) \
-		    /usr/share/pam/$configfile.md5sums 2>/dev/null
+		    "$DPKG_ROOT"/usr/share/pam/$configfile.md5sums 2>/dev/null
 		then
 			force=
 		fi
 	done
 fi
 
-pam-auth-update --package $force
+pam-auth-update --root "$DPKG_ROOT" --package $force
 
 if [ -n "$force" ]; then
-	rm -f /etc/pam.d/common-auth.pam-old \
-	      /etc/pam.d/common-account.pam-old \
-	      /etc/pam.d/common-password.pam-old \
-	      /etc/pam.d/common-session.pam-old
-elif dpkg --compare-versions "$2" lt-nl 1.1.0-1 \
-        && [ ! -e /etc/pam.d/common-session-noninteractive ]
-then
-	cp -a /etc/pam.d/common-session /etc/pam.d/common-session-noninteractive
+	rm -f "$DPKG_ROOT"/etc/pam.d/common-auth.pam-old \
+	      "$DPKG_ROOT"/etc/pam.d/common-account.pam-old \
+	      "$DPKG_ROOT"/etc/pam.d/common-password.pam-old \
+	      "$DPKG_ROOT"/etc/pam.d/common-session.pam-old
 fi
 
 
