--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/python3-dateutil.prerm	2019-02-06 08:35:10.000000000 -0500
+++ /media/mks/armbi_root/var/lib/dpkg/info/python3-dateutil.prerm	2023-03-17 12:05:45.000000000 -0400
@@ -1,11 +1,11 @@
 #!/bin/sh
 set -e
 
-# Automatically added by dh_python3:
-if which py3clean >/dev/null 2>&1; then
+# Automatically added by dh_python3
+if command -v py3clean >/dev/null 2>&1; then
 	py3clean -p python3-dateutil 
 else
-	dpkg -L python3-dateutil | perl -ne 's,/([^/]*)\.py$,/__pycache__/\1.*, or next; unlink $_ or die $! foreach glob($_)'
+	dpkg -L python3-dateutil | sed -En -e '/^(.*)\/(.+)\.py$/s,,rm "\1/__pycache__/\2".*,e'
 	find /usr/lib/python3/dist-packages/ -type d -name __pycache__ -empty -print0 | xargs --null --no-run-if-empty rmdir
 fi
 
