--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/sysfsutils.postinst	2018-11-28 21:11:47.000000000 -0500
+++ /media/mks/armbi_root/var/lib/dpkg/info/sysfsutils.postinst	2023-01-17 18:09:23.000000000 -0500
@@ -1,15 +1,15 @@
 #!/bin/sh
 set -e
-# Automatically added by dh_installinit/11.5.3
+# Automatically added by dh_installinit/13.11.4
 if [ "$1" = "configure" ] || [ "$1" = "abort-upgrade" ] || [ "$1" = "abort-deconfigure" ] || [ "$1" = "abort-remove" ] ; then
-	if [ -x "/etc/init.d/sysfsutils" ]; then
-		update-rc.d sysfsutils start 20 2 3 4 5 . >/dev/null
+	if [ -z "${DPKG_ROOT:-}" ] && [ -x "/etc/init.d/sysfsutils" ]; then
+		update-rc.d sysfsutils defaults >/dev/null
 		if [ -n "$2" ]; then
 			_dh_action=restart
 		else
 			_dh_action=start
 		fi
-		invoke-rc.d sysfsutils $_dh_action || exit 1
+		invoke-rc.d --skip-systemd-native sysfsutils $_dh_action || exit 1
 	fi
 fi
 # End automatically added section
