--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/bash.postrm	2019-04-18 00:12:36.000000000 -0400
+++ /media/mks/armbi_root/var/lib/dpkg/info/bash.postrm	2024-03-29 17:21:26.000000000 -0400
@@ -1,25 +1,5 @@
-#! /bin/sh
-
+#!/bin/sh
 set -e
-
-case "$1" in
-    upgrade|failed-upgrade|abort-install|abort-upgrade)
-	;;
-    remove|purge|disappear)
-	if which remove-shell >/dev/null && [ -f /etc/shells ]; then
-	    remove-shell /bin/bash
-	    remove-shell /bin/rbash
-	fi
-	;;
-    *)
-        echo "postrm called with unknown argument \`$1'" >&2
-        exit 1
-	;;
-esac
-
-# Automatically added by dh_installmenu/12.1.1
-if [ -x "`which update-menus 2>/dev/null`" ]; then update-menus ; fi
+# Automatically added by dh_installmenu/13.11.4
+if [ -x "`command -v update-menus`" ] && [ -x "$DPKG_ROOT`command -v update-menus`" ]; then update-menus ; fi
 # End automatically added section
-
-
-exit 0
