--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/bash.postinst	2019-04-18 00:12:36.000000000 -0400
+++ /media/mks/armbi_root/var/lib/dpkg/info/bash.postinst	2024-03-29 17:21:26.000000000 -0400
@@ -14,15 +14,8 @@
     10 \
   || true
 
-if [ "$1" = configure ] && dpkg --compare-versions "$2" le 3.2-3 \
-    && which add-shell >/dev/null
-then
-    add-shell /bin/bash
-    add-shell /bin/rbash
-fi
-
-# Automatically added by dh_installmenu/12.1.1
-if [ "$1" = "configure" ] && [ -x "`which update-menus 2>/dev/null`" ]; then
+# Automatically added by dh_installmenu/13.11.4
+if [ "$1" = "configure" ] && [ -x "`command -v update-menus`" ] && [ -x "$DPKG_ROOT`command -v update-menus`" ]; then
 	update-menus
 fi
 # End automatically added section
