--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/smartmontools.postrm	2018-10-15 16:03:35.000000000 -0400
+++ /media/mks/armbi_root/var/lib/dpkg/info/smartmontools.postrm	2022-10-15 20:09:15.000000000 -0400
@@ -10,30 +10,28 @@
 
 esac
 
-# Automatically added by dh_installinit/11.4.1
-if [ "$1" = "purge" ] ; then
+# Automatically added by dh_installinit/13.10
+if [ "$1" = "remove" ] && [ -x "/etc/init.d/smartmontools" ] ; then
+	chmod -x "/etc/init.d/smartmontools" >/dev/null || true
+fi
+if [ -z "${DPKG_ROOT:-}" ] && [ "$1" = "purge" ] ; then
 	update-rc.d smartmontools remove >/dev/null
 fi
 # End automatically added section
-# Automatically added by dh_installdeb/11.4.1
+# Automatically added by dh_installdeb/13.10
 dpkg-maintscript-helper rm_conffile /etc/smartd_warning.sh 6.3\+svn4002-2\~ -- "$@"
+dpkg-maintscript-helper rm_conffile /etc/init.d/smartd 6.1\+svn3812-1\~ -- "$@"
+dpkg-maintscript-helper rm_conffile /etc/smartmontools/run.d/10powersave-notify 7.0\~ -- "$@"
 # End automatically added section
-# Automatically added by dh_installsystemd/11.4.1
-if [ -d /run/systemd/system ]; then
+# Automatically added by dh_installsystemd/13.10
+if [ "$1" = remove ] && [ -d /run/systemd/system ] ; then
 	systemctl --system daemon-reload >/dev/null || true
 fi
 # End automatically added section
-# Automatically added by dh_installsystemd/11.4.1
-if [ "$1" = "remove" ]; then
-	if [ -x "/usr/bin/deb-systemd-helper" ]; then
-		deb-systemd-helper mask 'smartd.service' >/dev/null || true
-	fi
-fi
-
+# Automatically added by dh_installsystemd/13.10
 if [ "$1" = "purge" ]; then
 	if [ -x "/usr/bin/deb-systemd-helper" ]; then
-		deb-systemd-helper purge 'smartd.service' >/dev/null || true
-		deb-systemd-helper unmask 'smartd.service' >/dev/null || true
+		deb-systemd-helper purge 'smartmontools.service' >/dev/null || true
 	fi
 fi
 # End automatically added section
