--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/man-db.postrm	2019-02-10 07:14:20.000000000 -0500
+++ /media/mks/armbi_root/var/lib/dpkg/info/man-db.postrm	2023-03-12 18:23:59.000000000 -0400
@@ -17,49 +17,32 @@
 	fi
 	rm -f /var/lib/man-db/auto-update
 	;;
-
-    abort-install|abort-upgrade)
-	if [ -e /etc/cron.daily/man.moved-by-preinst ] && \
-	   [ ! -e /etc/cron.daily/man ]; then
-	    mv /etc/cron.daily/man.moved-by-preinst /etc/cron.daily/man
-	fi
-	if [ -e /etc/cron.weekly/catman.moved-by-preinst ] && \
-	   [ ! -e /etc/cron.weekly/catman ]; then
-	    mv /etc/cron.weekly/catman.moved-by-preinst /etc/cron.weekly/catman
-	fi
-	;;
 esac
 
-# Automatically added by dh_systemd_start/12
-if [ -d /run/systemd/system ]; then
+# Automatically added by dh_installsystemd/13.11.4
+if [ "$1" = remove ] && [ -d /run/systemd/system ] ; then
 	systemctl --system daemon-reload >/dev/null || true
 fi
 # End automatically added section
-# Automatically added by dh_systemd_enable/12
-if [ "$1" = "remove" ]; then
-	if [ -x "/usr/bin/deb-systemd-helper" ]; then
-		deb-systemd-helper mask 'man-db.timer' >/dev/null || true
-	fi
-fi
-
+# Automatically added by dh_installsystemd/13.11.4
 if [ "$1" = "purge" ]; then
 	if [ -x "/usr/bin/deb-systemd-helper" ]; then
 		deb-systemd-helper purge 'man-db.timer' >/dev/null || true
-		deb-systemd-helper unmask 'man-db.timer' >/dev/null || true
 	fi
 fi
 # End automatically added section
-# Automatically added by dh_installdebconf/12
+# Automatically added by dh_installdebconf/13.11.4
 if [ "$1" = purge ] && [ -e /usr/share/debconf/confmodule ]; then
 	. /usr/share/debconf/confmodule
 	db_purge
 fi
 # End automatically added section
-# Automatically added by dh_apparmor/2.13.2-7
+# Automatically added by dh_apparmor/3.0.8-3
 if [ "$1" = "purge" ] && ! [ -e "/etc/apparmor.d/usr.bin.man" ] ; then
     rm -f "/etc/apparmor.d/disable/usr.bin.man" || true
     rm -f "/etc/apparmor.d/force-complain/usr.bin.man" || true
     rm -f "/etc/apparmor.d/local/usr.bin.man" || true
+    rm -f /var/cache/apparmor/*/"usr.bin.man" || true
     rmdir /etc/apparmor.d/disable 2>/dev/null || true
     rmdir /etc/apparmor.d/local   2>/dev/null || true
     rmdir /etc/apparmor.d         2>/dev/null || true
