--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/python3-matplotlib.prerm	2018-12-20 19:00:46.000000000 -0500
+++ /media/mks/armbi_root/var/lib/dpkg/info/python3-matplotlib.prerm	2023-01-31 08:57:11.000000000 -0500
@@ -1,11 +1,11 @@
 #!/bin/sh
 set -e
 
-# Automatically added by dh_python3:
-if which py3clean >/dev/null 2>&1; then
-	py3clean -p python3-matplotlib 
+# Automatically added by dh_python3
+if command -v py3clean >/dev/null 2>&1; then
+	py3clean -p python3-matplotlib:arm64 
 else
-	dpkg -L python3-matplotlib | perl -ne 's,/([^/]*)\.py$,/__pycache__/\1.*, or next; unlink $_ or die $! foreach glob($_)'
+	dpkg -L python3-matplotlib:arm64 | sed -En -e '/^(.*)\/(.+)\.py$/s,,rm "\1/__pycache__/\2".*,e'
 	find /usr/lib/python3/dist-packages/ -type d -name __pycache__ -empty -print0 | xargs --null --no-run-if-empty rmdir
 fi
 
