--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/openssh-client.postinst	2020-01-31 15:55:34.000000000 -0500
+++ /media/mks/armbi_root/var/lib/dpkg/info/openssh-client.postinst	2024-06-22 15:38:08.000000000 -0400
@@ -2,46 +2,49 @@
 set -e
 
 action="$1"
-oldversion="$2"
 
 umask 022
 
 
-create_alternatives() {
-# Create alternatives for the various r* tools.
-# Make sure we don't change existing alternatives that a user might have
-# changed, but clean up after some old alternatives that mistakenly pointed
-# rlogin and rcp to ssh.
-	update-alternatives --quiet --remove rlogin /usr/bin/ssh
-	update-alternatives --quiet --remove rcp /usr/bin/ssh
-	for cmd in rsh rlogin rcp; do
-		scmd="s${cmd#r}"
-		if ! update-alternatives --display "$cmd" 2>/dev/null | \
-				grep -q "$scmd"; then
-			update-alternatives --quiet --install "/usr/bin/$cmd" "$cmd" "/usr/bin/$scmd" 20 \
-				--slave "/usr/share/man/man1/$cmd.1.gz" "$cmd.1.gz" "/usr/share/man/man1/$scmd.1.gz"
-		fi
-	done
+update_ssh_group_name() {
+	# The _ssh group used to be called ssh, but that could clash with
+	# locally-created user accounts.  Since this only exists as an
+	# otherwise-empty group to which ssh-agent is installed setgid, it's
+	# easy to rename.
+	if getent group ssh >/dev/null && ! getent group _ssh >/dev/null; then
+		groupmod -n _ssh ssh
+	fi
 }
 
 set_ssh_agent_permissions() {
-	if ! getent group ssh >/dev/null; then
-		addgroup --system --quiet ssh
+	if ! getent group _ssh >/dev/null; then
+		addgroup --system --quiet --force-badname _ssh
 	fi
 	if ! dpkg-statoverride --list /usr/bin/ssh-agent >/dev/null; then
-		chgrp ssh /usr/bin/ssh-agent
+		chgrp _ssh /usr/bin/ssh-agent
 		chmod 2755 /usr/bin/ssh-agent
 	fi
 }
 
+remove_obsolete_alternatives() {
+	update-alternatives --remove rcp /usr/bin/scp
+	update-alternatives --remove rlogin /usr/bin/slogin
+	update-alternatives --remove rsh /usr/bin/ssh
+}
+
 
 if [ "$action" = configure ]; then
-	create_alternatives
+	if dpkg --compare-versions "$2" lt-nl 1:8.4p1-6~; then
+		update_ssh_group_name
+	fi
 	set_ssh_agent_permissions
+	if dpkg --compare-versions "$2" lt-nl 1:9.1p1-1~; then
+		remove_obsolete_alternatives
+	fi
 fi
 
-# Automatically added by dh_installdeb/12.1.1
-dpkg-maintscript-helper rm_conffile /etc/ssh/moduli 1:7.9p1-8~ -- "$@"
+# Automatically added by dh_installdeb/13.11.4
+dpkg-maintscript-helper rm_conffile /etc/ssh/moduli 1:7.9p1-8\~ -- "$@"
 # End automatically added section
 
 
