--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/fontconfig-config.postrm	2018-11-05 15:42:38.000000000 -0500
+++ /media/mks/armbi_root/var/lib/dpkg/info/fontconfig-config.postrm	2023-01-31 17:24:08.000000000 -0500
@@ -43,7 +43,7 @@
 	    ;;
 esac
 
-# Automatically added by dh_installdeb/11.5.1
+# Automatically added by dh_installdeb/13.11.4
 dpkg-maintscript-helper rm_conffile /etc/fonts/conf.avail/10-autohint.conf 2.10 -- "$@"
 dpkg-maintscript-helper rm_conffile /etc/fonts/conf.avail/10-no-sub-pixel.conf 2.10 -- "$@"
 dpkg-maintscript-helper rm_conffile /etc/fonts/conf.avail/10-sub-pixel-bgr.conf 2.10 -- "$@"
@@ -76,7 +76,7 @@
 dpkg-maintscript-helper rm_conffile /etc/fonts/conf.avail/90-synthetic.conf 2.10 -- "$@"
 dpkg-maintscript-helper rm_conffile /etc/fonts/fonts.dtd 2.10 -- "$@"
 # End automatically added section
-# Automatically added by dh_installdebconf/11.5.1
+# Automatically added by dh_installdebconf/13.11.4
 if [ "$1" = purge ] && [ -e /usr/share/debconf/confmodule ]; then
 	. /usr/share/debconf/confmodule
 	db_purge
