--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/base-passwd.preinst	2019-02-10 14:10:55.000000000 -0500
+++ /media/mks/armbi_root/var/lib/dpkg/info/base-passwd.preinst	2022-09-20 06:22:51.000000000 -0400
@@ -2,8 +2,8 @@
 set -e
 
 if [ "$1" = install ]; then
-	if [ ! -e /etc/passwd ]; then
-		cat > /etc/passwd <<'EOF'
+	if [ ! -e "$DPKG_ROOT/etc/passwd" ]; then
+		cat > "$DPKG_ROOT/etc/passwd" <<'EOF'
 root:*:0:0:root:/root:/bin/bash
 daemon:*:1:1:daemon:/usr/sbin:/usr/sbin/nologin
 bin:*:2:2:bin:/bin:/usr/sbin/nologin
@@ -19,14 +19,14 @@
 www-data:*:33:33:www-data:/var/www:/usr/sbin/nologin
 backup:*:34:34:backup:/var/backups:/usr/sbin/nologin
 list:*:38:38:Mailing List Manager:/var/list:/usr/sbin/nologin
-irc:*:39:39:ircd:/var/run/ircd:/usr/sbin/nologin
-gnats:*:41:41:Gnats Bug-Reporting System (admin):/var/lib/gnats:/usr/sbin/nologin
+irc:*:39:39:ircd:/run/ircd:/usr/sbin/nologin
+_apt:*:42:65534::/nonexistent:/usr/sbin/nologin
 nobody:*:65534:65534:nobody:/nonexistent:/usr/sbin/nologin
 EOF
 	fi
 
-	if [ ! -e /etc/group ]; then
-		cat > /etc/group <<'EOF'
+	if [ ! -e "$DPKG_ROOT/etc/group" ]; then
+		cat > "$DPKG_ROOT/etc/group" <<'EOF'
 root:*:0:
 daemon:*:1:
 bin:*:2:
@@ -56,7 +56,6 @@
 list:*:38:
 irc:*:39:
 src:*:40:
-gnats:*:41:
 shadow:*:42:
 utmp:*:43:
 video:*:44:
