--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/pciutils.postinst	2016-11-30 01:53:07.000000000 -0500
+++ /media/mks/armbi_root/var/lib/dpkg/info/pciutils.postinst	2023-01-17 18:50:53.000000000 -0500
@@ -1,21 +1,21 @@
-#! /bin/sh
+#!/bin/sh
 # postinst script for pciutils
 
 set -e
 
 case "$1" in
 configure)
-	for ext in new neww old
-	do
-		rm -f /usr/share/misc/pci.ids.$ext
-	done
-	;;
+  for ext in new neww old
+  do
+    rm -f /usr/share/misc/pci.ids.$ext
+  done
+  ;;
 abort-upgrade|abort-remove|abort-deconfigure)
-	;;
+  ;;
 *)
-	echo "postinst called with unknown argument \`$1'" >&2
-	exit 1
-	;;
+  echo "postinst called with unknown argument '$1'" >&2
+  exit 1
+  ;;
 esac
 
 
