--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/initramfs-tools.postinst	2019-08-22 21:16:37.000000000 -0400
+++ /media/mks/armbi_root/var/lib/dpkg/info/initramfs-tools.postinst	2022-07-12 17:51:34.000000000 -0400
@@ -44,10 +44,8 @@
 	DPKG_MAINTSCRIPT_PACKAGE='' update-initramfs -u
 fi
 
-# Automatically added by dh_installdeb/12.1.1
-dpkg-maintscript-helper rm_conffile /etc/bash_completion.d/initramfs-tools 0.126~ -- "$@"
-# End automatically added section
-# Automatically added by dh_installdeb/12.1.1
-dpkg-maintscript-helper rm_conffile /etc/kernel/signed_postinst.d/initramfs-tools 0.126~ -- "$@"
+# Automatically added by dh_installdeb/13.8
+dpkg-maintscript-helper rm_conffile /etc/bash_completion.d/initramfs-tools 0.126\~ -- "$@"
+dpkg-maintscript-helper rm_conffile /etc/kernel/signed_postinst.d/initramfs-tools 0.126\~ -- "$@"
 # End automatically added section
 
