--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/smartmontools.postinst	2018-10-15 16:03:35.000000000 -0400
+++ /media/mks/armbi_root/var/lib/dpkg/info/smartmontools.postinst	2022-10-15 20:09:15.000000000 -0400
@@ -4,41 +4,44 @@
 # Remove shutdown and reboot links; this init script does not need them.
 case "$1" in
   configure)
-	if dpkg --compare-versions "$2" lt "5.38+svn2879-1"; then
-		if [ -e /etc/rc0.d/K20smartmontools ]; then
-			rm -f /etc/rc0.d/K20smartmontools
-		fi
-		if [ -e /etc/rc6.d/K20smartmontools ]; then
-			rm -f /etc/rc6.d/K20smartmontools
-		fi
-	fi
- ;;
+    ## https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=976696
+    ## TODO: As suggested in #976696, it would be better to avoid replacing
+    ## newer "drivedb.h" unconditionally if user updated it with
+    ## "update-smart-drivedb".
+    cp -f /usr/share/smartmontools/drivedb.h  /var/lib/smartmontools/drivedb/
+  ;;
 esac
 
-dpkg-maintscript-helper rm_conffile \
-	/etc/init.d/smartd 6.1+svn3812-1~ smartmontools -- "$@"
-
-# Automatically added by dh_installdeb/11.4.1
+# Automatically added by dh_installdeb/13.10
 dpkg-maintscript-helper rm_conffile /etc/smartd_warning.sh 6.3\+svn4002-2\~ -- "$@"
+dpkg-maintscript-helper rm_conffile /etc/init.d/smartd 6.1\+svn3812-1\~ -- "$@"
+dpkg-maintscript-helper rm_conffile /etc/smartmontools/run.d/10powersave-notify 7.0\~ -- "$@"
+# End automatically added section
+# Automatically added by dh_installinit/13.10
+if [ "$1" = "configure" ] || [ "$1" = "abort-upgrade" ] || [ "$1" = "abort-deconfigure" ] || [ "$1" = "abort-remove" ] ; then
+	if [ -x "/etc/init.d/smartmontools" ]; then
+		update-rc.d smartmontools defaults >/dev/null || exit 1
+	fi
+fi
 # End automatically added section
-# Automatically added by dh_installsystemd/11.4.1
+# Automatically added by dh_installsystemd/13.10
 if [ "$1" = "configure" ] || [ "$1" = "abort-upgrade" ] || [ "$1" = "abort-deconfigure" ] || [ "$1" = "abort-remove" ] ; then
-	# This will only remove masks created by d-s-h on package removal.
-	deb-systemd-helper unmask 'smartd.service' >/dev/null || true
+	# The following line should be removed in trixie or trixie+1
+	deb-systemd-helper unmask 'smartmontools.service' >/dev/null || true
 
 	# was-enabled defaults to true, so new installations run enable.
-	if deb-systemd-helper --quiet was-enabled 'smartd.service'; then
+	if deb-systemd-helper --quiet was-enabled 'smartmontools.service'; then
 		# Enables the unit on first installation, creates new
 		# symlinks on upgrades if the unit file has changed.
-		deb-systemd-helper enable 'smartd.service' >/dev/null || true
+		deb-systemd-helper enable 'smartmontools.service' >/dev/null || true
 	else
 		# Update the statefile to add new symlinks (if any), which need to be
 		# cleaned up on purge. Also remove old symlinks.
-		deb-systemd-helper update-state 'smartd.service' >/dev/null || true
+		deb-systemd-helper update-state 'smartmontools.service' >/dev/null || true
 	fi
 fi
 # End automatically added section
-# Automatically added by dh_installsystemd/11.4.1
+# Automatically added by dh_installsystemd/13.10
 if [ "$1" = "configure" ] || [ "$1" = "abort-upgrade" ] || [ "$1" = "abort-deconfigure" ] || [ "$1" = "abort-remove" ] ; then
 	if [ -d /run/systemd/system ]; then
 		systemctl --system daemon-reload >/dev/null || true
@@ -47,20 +50,7 @@
 		else
 			_dh_action=start
 		fi
-		deb-systemd-invoke $_dh_action 'smartd.service' >/dev/null || true
-	fi
-fi
-# End automatically added section
-# Automatically added by dh_installinit/11.4.1
-if [ "$1" = "configure" ] || [ "$1" = "abort-upgrade" ] || [ "$1" = "abort-deconfigure" ] || [ "$1" = "abort-remove" ] ; then
-	if [ -x "/etc/init.d/smartmontools" ]; then
-		update-rc.d smartmontools defaults >/dev/null
-		if [ -n "$2" ]; then
-			_dh_action=restart
-		else
-			_dh_action=start
-		fi
-		invoke-rc.d smartmontools $_dh_action || exit 1
+		deb-systemd-invoke $_dh_action 'smartmontools.service' >/dev/null || true
 	fi
 fi
 # End automatically added section
