--- /media/mks/5dcae443-0796-4029-ae6c-bf5bd2a37067/var/lib/dpkg/info/screen.preinst	2021-02-20 15:59:38.000000000 -0500
+++ /media/mks/armbi_root/var/lib/dpkg/info/screen.preinst	2023-01-08 22:56:37.000000000 -0500
@@ -1,14 +1,7 @@
 #!/bin/sh
-
 set -e
-
-if [ "$1" = upgrade ]; then
-  if dpkg --compare-versions "$2" lt-nl 4.0.3-7 && \
-     dpkg --compare-versions "$2" gt 4.0.3-3; then
-    if ! test -L /tmp/.screen && test -d /tmp/.screen && test -n "`find /tmp/.screen -type p 2>/dev/null`"; then
-      ln -s /tmp/.screen /run/screen
-    fi
-  fi
+# Automatically added by dh_installinit/13.11.4
+if [ "$1" = "install" ] && [ -n "$2" ] && [ -e "/etc/init.d/screen-cleanup" ] ; then
+	chmod +x "/etc/init.d/screen-cleanup" >/dev/null || true
 fi
-
-
+# End automatically added section
